// @neynar/react/dist/bundle.es.js
import { styled as g } from "@pigment-css/react";
import V, { useState as N, createContext as ae, useContext as ce, useEffect as H, useMemo as f1, useRef as Y, useCallback as J, useLayoutEffect as xt, useDebugValue as wt, memo as He } from "react";
import M1 from "hls.js";
const s1 = {
  vars: {
    colors: {
      primary: "var(--colors-primary)"
    },
    typography: {
      fonts: {
        base: "var(--typography-fonts-base)"
      },
      fontWeights: {
        regular: "var(--typography-fontWeights-regular)",
        bold: "var(--typography-fontWeights-bold)"
      },
      fontSizes: {
        large: "var(--typography-fontSizes-large)",
        medium: "var(--typography-fontSizes-medium)",
        small: "var(--typography-fontSizes-small)"
      }
    },
    palette: {
      background: "var(--palette-background)",
      border: "var(--palette-border)",
      text: "var(--palette-text)",
      textMuted: "var(--palette-textMuted)"
    }
  }
};
var Ue = { exports: {} }, y1 = {};
/**
 * @license React
 * react-jsx-runtime.production.js
 *
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var me;
function vt() {
  if (me) return y1;
  me = 1;
  var e = Symbol.for("react.transitional.element"), t = Symbol.for("react.fragment");
  function s(o, i, r) {
    var a = null;
    if (r !== void 0 && (a = "" + r), i.key !== void 0 && (a = "" + i.key), "key" in i) {
      r = {};
      for (var u in i)
        u !== "key" && (r[u] = i[u]);
    } else r = i;
    return i = r.ref, {
      $$typeof: e,
      type: o,
      key: a,
      ref: i !== void 0 ? i : null,
      props: r
    };
  }
  return y1.Fragment = t, y1.jsx = s, y1.jsxs = s, y1;
}
Ue.exports = vt();
var n = Ue.exports, Pe = /* @__PURE__ */ ((e) => (e.Light = "light", e.Dark = "dark", e))(Pe || {}), c1 = /* @__PURE__ */ ((e) => (e.FARCASTER = "farcaster", e.NEYNAR = "neynar", e.WARPCAST = "warpcast", e))(c1 || {});
function le(e, t = "", {
  serialize: s = JSON.stringify,
  deserialize: o = JSON.parse
} = {}) {
  const [i, r] = N(() => {
    if (typeof window > "u")
      return t;
    try {
      const d = window.localStorage.getItem(e);
      return d ? o(d) : t;
    } catch (d) {
      return console.error("Error reading from localStorage", d), t;
    }
  });
  return [i, (d) => {
    try {
      const l = d instanceof Function ? d(i) : d;
      r(l), typeof window < "u" && window.localStorage.setItem(e, s(l));
    } catch (l) {
      console.error("Error writing to localStorage", l);
    }
  }, () => {
    try {
      window.localStorage.removeItem(e), r(t);
    } catch (d) {
      console.error("Error removing from localStorage", d);
    }
  }];
}
var w1 = /* @__PURE__ */ ((e) => (e.NEYNAR_AUTHENTICATED_USER = "neynar_authenticated_user", e))(w1 || {});
const qe = ae(void 0), jt = ({
  children: e,
  _setIsAuthenticated: t,
  _setUser: s,
  _onAuthSuccess: o,
  _onSignout: i
}) => {
  const {
    isAuthenticated: r
  } = o1(), [a, u] = N(!1), [d, l] = N(null), [m] = le(w1.NEYNAR_AUTHENTICATED_USER);
  H(() => {
    t(a);
  }, [a]), H(() => {
    u(r);
  }, [r]), H(() => {
    m ? (l(m), u(!0)) : (l(null), u(!1));
  }, []), H(() => {
    s(d);
  }, [d]);
  const f = (x) => {
    o && o(x);
  }, C = (x) => {
    i && i(x);
  }, c = f1(() => ({
    isAuthenticated: a,
    user: d,
    setIsAuthenticated: u,
    setUser: l,
    onAuthSuccess: f,
    onSignout: C
  }), [a, d]);
  return /* @__PURE__ */ n.jsx(qe.Provider, {
    value: c,
    children: e
  });
}, We = () => {
  const e = ce(qe);
  if (!e)
    throw new Error("useAuth must be used within a AuthContextProvider");
  return e;
}, yt = /* @__PURE__ */ g("div")({
  classes: ["tbaom7c"]
}), Et = /* @__PURE__ */ g("div")({
  classes: ["t1ff44zi"],
  variants: [{
    props: {
      type: "success"
    },
    className: "t1ff44zi-1"
  }, {
    props: {
      type: "error"
    },
    className: "t1ff44zi-2"
  }, {
    props: {
      type: "warning"
    },
    className: "t1ff44zi-3"
  }, {
    props: {
      type: "info"
    },
    className: "t1ff44zi-4"
  }]
});
var Be = /* @__PURE__ */ ((e) => (e.Success = "success", e.Error = "error", e.Warning = "warning", e.Info = "info", e))(Be || {});
const ze = ae(void 0), pr = ({
  children: e,
  settings: {
    clientId: t,
    defaultTheme: s = Pe.Light,
    eventsCallbacks: o
  }
}) => {
  const [i] = N(t), [r, a] = N(!1), [u, d] = N(s), [l, m] = N([]), [f, C] = N(null), c = (E, R) => {
    const h = {
      type: E,
      message: R
    };
    m((w) => [...w, h]), setTimeout(() => x(h), 5e3);
  }, x = (E) => {
    m((R) => R.filter((h) => h !== E));
  };
  H(() => {
    const E = document.querySelector(":root");
    E && (u === "light" ? (E.classList.add("theme-light"), E.classList.remove("theme-dark")) : u === "dark" && (E.classList.add("theme-dark"), E.classList.remove("theme-light")));
  }, [u]);
  const p = (E) => {
    a(E);
  }, y = (E) => {
    C(E);
  }, v = () => {
    if (f) {
      const {
        signer_uuid: E,
        ...R
      } = f;
      C(null), a(!1), localStorage.removeItem(w1.NEYNAR_AUTHENTICATED_USER), o != null && o.onSignout && o.onSignout(R);
    }
  }, S = f1(() => ({
    client_id: i,
    theme: u,
    isAuthenticated: r,
    user: f,
    setTheme: d,
    showToast: c,
    logoutUser: v
  }), [i, u, r, f, d, c, v]);
  return /* @__PURE__ */ n.jsx(ze.Provider, {
    value: S,
    children: /* @__PURE__ */ n.jsxs(jt, {
      _setIsAuthenticated: p,
      _setUser: y,
      _onAuthSuccess: o == null ? void 0 : o.onAuthSuccess,
      _onSignout: o == null ? void 0 : o.onSignout,
      children: [e, /* @__PURE__ */ n.jsx(yt, {
        children: l.map((E, R) => /* @__PURE__ */ n.jsx(Et, {
          type: E.type,
          children: E.message
        }, R))
      })]
    })
  });
}, o1 = () => {
  const e = ce(ze);
  if (!e)
    throw new Error("useNeynarContext must be used within a NeynarContextProvider");
  return e;
}, _t = () => /* @__PURE__ */ n.jsxs("svg", {
  width: "18",
  height: "17",
  viewBox: "0 0 18 17",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  children: [/* @__PURE__ */ n.jsx("path", {
    d: "M13.7222 0H4.27778C2.19137 0 0.5 1.69137 0.5 3.77778V13.2222C0.5 15.3086 2.19137 17 4.27778 17H13.7222C15.8086 17 17.5 15.3086 17.5 13.2222V3.77778C17.5 1.69137 15.8086 0 13.7222 0Z",
    fill: "#855DCD"
  }), /* @__PURE__ */ n.jsx("path", {
    d: "M4.88281 2.64453H13.1184V14.3556H11.9095V8.9912H11.8976C11.764 7.50859 10.518 6.34675 9.00059 6.34675C7.4832 6.34675 6.23717 7.50859 6.10356 8.9912H6.0917V14.3556H4.88281V2.64453Z",
    fill: "white"
  }), /* @__PURE__ */ n.jsx("path", {
    d: "M2.69141 4.30664L3.18252 5.96886H3.59807V12.6933C3.38943 12.6933 3.2203 12.8625 3.2203 13.0711V13.5244H3.14474C2.9361 13.5244 2.76696 13.6936 2.76696 13.9022V14.3555H6.99807V13.9022C6.99807 13.6936 6.82893 13.5244 6.6203 13.5244H6.54474V13.0711C6.54474 12.8625 6.3756 12.6933 6.16696 12.6933H5.71363V4.30664H2.69141Z",
    fill: "white"
  }), /* @__PURE__ */ n.jsx("path", {
    d: "M11.9854 12.6933C11.7768 12.6933 11.6076 12.8625 11.6076 13.0711V13.5244H11.5321C11.3235 13.5244 11.1543 13.6936 11.1543 13.9022V14.3555H15.3854V13.9022C15.3854 13.6936 15.2162 13.5244 15.0076 13.5244H14.9321V13.0711C14.9321 12.8625 14.7629 12.6933 14.5543 12.6933V5.96886H14.9699L15.461 4.30664H12.4387V12.6933H11.9854Z",
    fill: "white"
  })]
}), xe = () => /* @__PURE__ */ n.jsxs("svg", {
  width: "32",
  height: "17",
  viewBox: "0 0 50 28",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  children: [/* @__PURE__ */ n.jsx("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M38.3744 13.7311C38.3744 16.1209 37.7639 18.368 36.6904 20.3252C32.5685 19.8525 27.847 18.9503 22.8999 17.6247C20.4058 16.9564 18.0151 16.2236 15.7697 15.4495L15.7693 15.4486C15.7579 15.445 15.7465 15.4413 15.7351 15.4376C15.6448 15.4064 15.5547 15.3752 15.4649 15.3439C12.8978 14.399 11.2356 12.2056 11.4168 10.0315L11.4482 10.0399L11.4215 10.0133C13.0427 4.23612 18.3483 0 24.6434 0C32.2268 0 38.3744 6.14762 38.3744 13.7311Z",
    fill: "black"
  }), /* @__PURE__ */ n.jsx("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M11.3804 17.2987C12.4882 16.5313 13.9658 16.1804 15.5002 16.3953C17.7731 17.156 20.1682 17.8782 22.6535 18.5441C27.397 19.8151 31.9584 20.7444 36.0833 21.3275C33.623 25.0252 29.4177 27.4619 24.6434 27.4619C18.294 27.4619 12.9511 23.1524 11.3804 17.2987Z",
    fill: "black"
  }), /* @__PURE__ */ n.jsx("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M43.0917 13.4458C41.6461 12.6648 39.9756 11.8759 38.1233 11.1029C38.0363 10.6544 37.9275 10.2136 37.7981 9.78174C39.9806 10.6648 41.9403 11.5752 43.6139 12.4794C45.7007 13.6069 47.3703 14.7394 48.4714 15.816C49.5356 16.8566 50.2295 18.0063 49.93 19.1241C49.6304 20.2419 48.4547 20.8906 47.0128 21.2597C45.5209 21.6415 43.5087 21.7875 41.1377 21.7206C39.6046 21.6773 37.9082 21.5444 36.0864 21.3234C36.3086 20.9891 36.5165 20.6446 36.7094 20.2906C38.3134 20.4733 39.8091 20.5841 41.1687 20.6225C43.4946 20.6882 45.3908 20.5409 46.7404 20.1955C48.14 19.8373 48.7389 19.3249 48.8689 18.8398C48.9989 18.3547 48.7364 17.6115 47.7034 16.6015C46.7073 15.6275 45.1388 14.5519 43.0917 13.4458ZM8.86229 3.16772C10.7583 3.22128 12.9038 3.41177 15.2278 3.73661C14.8974 4.04803 14.5824 4.37566 14.2841 4.71822C12.3016 4.46314 10.4678 4.312 8.83127 4.26577C6.50541 4.20006 4.6092 4.34737 3.25958 4.69279C1.86001 5.05099 1.26107 5.56342 1.13109 6.04849C1.00112 6.53356 1.2636 7.27681 2.29657 8.2868C3.29267 9.26075 4.86118 10.3364 6.90828 11.4425C6.94181 11.4606 6.97546 11.4787 7.00923 11.4968C6.84251 11.5938 6.72574 11.7161 6.66353 11.8645C6.57791 12.0688 6.60013 12.3121 6.72013 12.5874C6.60757 12.5279 6.49623 12.4684 6.38612 12.4089C4.29926 11.2814 2.62971 10.1489 1.5286 9.07223C0.464367 8.03166 -0.229485 6.882 0.0700354 5.76418C0.369556 4.64635 1.54528 3.99764 2.98722 3.6286C4.47911 3.24677 6.49126 3.10074 8.86229 3.16772Z",
    fill: "black"
  }), /* @__PURE__ */ n.jsx("ellipse", {
    cx: "5.94866",
    cy: "11.5309",
    rx: "3.76009",
    ry: "0.575832",
    transform: "rotate(29.2824 5.94866 11.5309)",
    fill: "black"
  })]
}), Rt = () => /* @__PURE__ */ n.jsxs("svg", {
  width: "17",
  height: "17",
  viewBox: "0 0 17 17",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  children: [/* @__PURE__ */ n.jsx("mask", {
    id: "mask0_875_48",
    maskUnits: "userSpaceOnUse",
    x: "0",
    y: "0",
    width: "17",
    height: "17",
    children: /* @__PURE__ */ n.jsx("path", {
      d: "M16.9947 0H0V16.9947H16.9947V0Z",
      fill: "white"
    })
  }), /* @__PURE__ */ n.jsxs("g", {
    mask: "url(#mask0_875_48)",
    children: [/* @__PURE__ */ n.jsx("path", {
      d: "M12.7871 16.9947H4.20765C1.88755 16.9947 0 15.1073 0 12.7871V4.20771C0 1.88756 1.88755 0 4.20765 0H12.7871C15.1072 0 16.9947 1.88756 16.9947 4.20771V12.7871C16.9947 15.1073 15.1072 16.9947 12.7871 16.9947Z",
      fill: "#472A91"
    }), /* @__PURE__ */ n.jsx("path", {
      d: "M11.1523 5.37891L10.3143 8.52601L9.47364 5.37891H7.53893L6.69017 8.54919L5.84419 5.37891H3.64062L5.68823 12.3371H7.58931L8.49835 9.10599L9.40736 12.3371H11.3125L13.3557 5.37891H11.1523Z",
      fill: "white"
    })]
  })]
});
var St = {};
const G1 = /* @__PURE__ */ g("img")({
  classes: ["ippfsqr"]
}), At = /* @__PURE__ */ g("button")({
  classes: ["b1yzssjb"]
}), bt = /* @__PURE__ */ g("div")({
  classes: ["m1xn8b2e"]
}), Lt = /* @__PURE__ */ g("button")({
  classes: ["mwuiqyg"]
}), It = (e, t) => {
  if (t)
    return t;
  switch (e) {
    case c1.FARCASTER:
      return "Sign in with Farcaster";
    case c1.NEYNAR:
      return "Sign in with Neynar";
    case c1.WARPCAST:
      return "Sign in with Warpcast";
    default:
      return "Sign in with Neynar";
  }
}, Tt = (e, t, s) => {
  if (t)
    return t;
  if (s)
    return /* @__PURE__ */ n.jsx(G1, {
      src: s,
      alt: "Custom logo"
    });
  switch (e) {
    case c1.FARCASTER:
      return /* @__PURE__ */ n.jsx(_t, {});
    case c1.NEYNAR:
      return /* @__PURE__ */ n.jsx(xe, {});
    case c1.WARPCAST:
      return /* @__PURE__ */ n.jsx(Rt, {});
    default:
      return /* @__PURE__ */ n.jsx(xe, {});
  }
}, Vt = ({
  children: e,
  label: t,
  variant: s = c1.NEYNAR,
  icon: o,
  customLogoUrl: i,
  modalStyle: r = {},
  modalButtonStyle: a = {},
  ...u
}) => {
  const {
    client_id: d,
    user: l,
    isAuthenticated: m
  } = o1(), {
    setIsAuthenticated: f,
    setUser: C,
    onAuthSuccess: c,
    onSignout: x
  } = We(), [p, y, v] = le(w1.NEYNAR_AUTHENTICATED_USER), [S, E] = N(!1), R = Y(null), h = `${St.NEYNAR_LOGIN_URL ?? "https://app.neynar.com/login"}?client_id=${d}`, w = new URL(h).origin, I = Y(null), j = J(async (L) => {
    var P;
    if (L.origin === w && L.data && L.data.is_authenticated) {
      f(!0), (P = R.current) == null || P.close(), window.removeEventListener("message", j);
      const q = {
        signer_uuid: L.data.signer_uuid,
        ...L.data.user
      };
      y(q), C(q), c({
        user: q
      });
    }
  }, [d, f]), T = J(() => {
    const q = window.screen.width / 2 - 300, b = `width=600,height=700,top=${window.screen.height / 2 - 700 / 2},left=${q}`;
    if (R.current = window.open(h, "_blank", b), !R.current) {
      console.error("Failed to open the authentication window. Please check your pop-up blocker settings.");
      return;
    }
    window.addEventListener("message", j, !1);
  }, [d, j]), A = () => {
    if (l) {
      const L = l;
      v(), f(!1), z();
      const {
        signer_uuid: P,
        ...q
      } = L;
      x(q);
    }
  }, F = () => E(!0), z = () => E(!1);
  H(() => () => {
    window.removeEventListener("message", j);
  }, [j]);
  const D = J((L) => {
    I.current && !I.current.contains(L.target) && z();
  }, []);
  return H(() => (S ? document.addEventListener("mousedown", D) : document.removeEventListener("mousedown", D), () => {
    document.removeEventListener("mousedown", D);
  }), [S, D]), /* @__PURE__ */ n.jsxs(n.Fragment, {
    children: [S && /* @__PURE__ */ n.jsxs(bt, {
      style: r,
      ref: I,
      children: [/* @__PURE__ */ n.jsx(G1, {
        src: l == null ? void 0 : l.pfp_url,
        alt: l == null ? void 0 : l.username
      }), /* @__PURE__ */ n.jsxs("span", {
        children: ["@", l == null ? void 0 : l.username]
      }), /* @__PURE__ */ n.jsx(Lt, {
        style: a,
        onClick: A,
        children: "Sign out"
      })]
    }), /* @__PURE__ */ n.jsx(At, {
      onClick: m ? F : T,
      ...u,
      children: m ? /* @__PURE__ */ n.jsxs(n.Fragment, {
        children: [/* @__PURE__ */ n.jsx(G1, {
          src: l == null ? void 0 : l.pfp_url,
          alt: `${l == null ? void 0 : l.username} profile picture`
        }), /* @__PURE__ */ n.jsxs("span", {
          style: {
            marginLeft: "10px"
          },
          children: ["@", l == null ? void 0 : l.username]
        })]
      }) : /* @__PURE__ */ n.jsxs(n.Fragment, {
        children: [Tt(s, o, i), /* @__PURE__ */ n.jsx("span", {
          style: {
            marginLeft: "5px"
          },
          children: It(s, t)
        })]
      })
    })]
  });
}, N1 = /* @__PURE__ */ g("img")({
  classes: ["a11pt0xp"],
  vars: {
    "a11pt0xp-0": [(e, t = s1) => e.width || "45px", !1],
    "a11pt0xp-1": [(e, t = s1) => e.width || "45px", !1]
  }
}), we = "https://warpcast.com", Ze = /(^|\s)\/\w+/g, Ye = /@\w+(\.eth)?/g, Ge = /((https?:\/\/)?([a-zA-Z0-9.-]+\.[a-zA-Z]{2,})(\/[^\s]*)?)/g, ve = new RegExp(`(${Ze.source})|(${Ye.source})|(${Ge.source})`, "g"), Nt = (e) => Ze.test(e) ? `${we}/~/channel${e.trim()}` : Ye.test(e) ? `${we}/${e.substring(1)}` : Ge.test(e) ? e.startsWith("http") ? e : `http://${e}` : "", $t = /* @__PURE__ */ g("a")({
  classes: ["s4xymqs"]
}), kt = (e) => e.filter((t) => t.url).map((t) => t.url), Dt = (e, t) => {
  if (!e) return [];
  const s = kt(t), o = [];
  let i = 0, r;
  for (; (r = ve.exec(e)) !== null; ) {
    const a = r.index;
    i < a && o.push(e.slice(i, a));
    const u = r[0].trim();
    if (s.includes(u))
      o.push(u);
    else {
      const d = Nt(u);
      o.push(/* @__PURE__ */ n.jsx($t, {
        href: d,
        target: "_blank",
        children: u
      }, a));
    }
    i = ve.lastIndex;
  }
  return i < e.length && o.push(e.slice(i)), o;
}, K = /* @__PURE__ */ g("div")({
  classes: ["be87m0m"],
  vars: {
    "be87m0m-0": [(e, t = s1) => e.alignItems || "flex-start", !1],
    "be87m0m-1": [(e, t = s1) => e.justifyContent || "flex-start", !1],
    "be87m0m-2": [(e, t = s1) => e.flexGrow || "initial", !0],
    "be87m0m-3": [(e, t = s1) => e.flexShrink || "initial", !0],
    "be87m0m-4": [(e, t = s1) => e.spacing ?? e.spacingVertical ?? e.spacingTop ?? "0px", !1],
    "be87m0m-5": [(e, t = s1) => e.spacing ?? e.spacingHorizontal ?? e.spacingRight ?? "0px", !1],
    "be87m0m-6": [(e, t = s1) => e.spacing ?? e.spacingVertical ?? e.spacingBottom ?? "0px", !1],
    "be87m0m-7": [(e, t = s1) => e.spacing ?? e.spacingHorizontal ?? e.spacingLeft ?? "0px", !1]
  }
}), Ft = () => K, Je = /* @__PURE__ */ g(Ft())({
  classes: ["v1lr8cv7"]
}), Mt = () => K, X = /* @__PURE__ */ g(Mt())({
  classes: ["ha1dvpi"]
}), Ke = () => /* @__PURE__ */ n.jsxs("svg", {
  width: "18",
  height: "18",
  viewBox: "0 0 18 18",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  children: [/* @__PURE__ */ n.jsx("rect", {
    width: "18",
    height: "18",
    rx: "9",
    fill: "#8A63D2"
  }), /* @__PURE__ */ n.jsx("path", {
    d: "M13.375 7.19002H10.25V3.58541C10.25 3.40518 10.125 3.22495 10 3.10479C9.75 2.92456 9.3125 2.98464 9.125 3.22495L4.125 9.83339C4.0625 9.95355 4 10.0737 4 10.1939C4 10.5543 4.25 10.7946 4.625 10.7946H7.75V14.3992C7.75 14.7597 8 15 8.375 15C8.5625 15 8.75 14.8798 8.875 14.7597L13.875 8.15125C13.9375 8.03109 14 7.91094 14 7.79078C14 7.43032 13.75 7.19002 13.375 7.19002Z",
    fill: "white"
  })]
}), C1 = "https://sdk-api.neynar.com", Ot = "https://api.allorigins.win/get", E1 = "https://i.imgur.com/gxH8azy.png";
var Ht = {
  SDK_VERSION: '"1.2.0"'
};
const Ut = Ht.SDK_VERSION, p1 = async (e, t = {}) => (t.headers = {
  ...t.headers,
  "x-sdk": "react",
  "x-sdk-version": Ut
}, fetch(e, t));
async function Pt({
  type: e,
  identifier: t,
  viewerFid: s,
  client_id: o
}) {
  try {
    let i = `${C1}/v2/farcaster/cast?type=${e}&identifier=${t}${s ? `&viewer_fid=${s}` : ""}&client_id=${o}`;
    const a = await (await p1(i)).json();
    return (a == null ? void 0 : a.cast) || null;
  } catch (i) {
    return console.error("Error fetching cast by identifier", i), null;
  }
}
const qt = ({
  type: e,
  identifier: t,
  viewerFid: s,
  allowReactions: o = !1,
  renderEmbeds: i = !0,
  renderFrames: r = !1,
  onLikeBtnPress: a,
  onRecastBtnPress: u,
  onCommentBtnPress: d,
  onFrameBtnPress: l,
  containerStyles: m,
  textStyles: f
}) => {
  const {
    client_id: C
  } = o1(), [c, x] = V.useState(null), [p, y] = V.useState(!0), [v, S] = V.useState(null), E = (c == null ? void 0 : c.author.fid) === s;
  return V.useEffect(() => {
    e && t && (y(!0), S(null), Pt({
      type: e,
      identifier: t,
      viewerFid: s,
      client_id: C
    }).then((R) => {
      x(R);
    }).catch((R) => {
      S(R);
    }).finally(() => {
      y(!1);
    }));
  }, [e, t, s, C]), p ? /* @__PURE__ */ n.jsx("div", {
    children: "Loading..."
  }) : !c || v ? /* @__PURE__ */ n.jsx("div", {
    children: "Error: could not fetch cast data"
  }) : r && !l ? /* @__PURE__ */ n.jsx("div", {
    children: "Error: onFrameBtnPress must be provided when renderEmbeds is true."
  }) : /* @__PURE__ */ n.jsx($1, {
    username: c.author.username,
    displayName: c.author.display_name,
    avatarImgUrl: c.author.pfp_url,
    text: c.text,
    hash: c.hash,
    reactions: c.reactions,
    replies: c.replies.count,
    embeds: c.embeds ?? [],
    frames: c.frames ?? [],
    renderEmbeds: i,
    renderFrames: r,
    channel: c.channel ? {
      id: c.channel.id,
      name: c.channel.name,
      url: c.parent_url
    } : void 0,
    viewerFid: s,
    allowReactions: o,
    hasPowerBadge: c.author.power_badge,
    isOwnProfile: E,
    containerStyles: m,
    textStyles: f,
    onLikeBtnPress: a,
    onRecastBtnPress: u,
    onCommentBtnPress: d,
    onFrameBtnPress: l
  });
}, Wt = /* @__PURE__ */ g("a")({
  classes: ["s1uo4lx4"]
}), O1 = /* @__PURE__ */ new Map(), b1 = /* @__PURE__ */ new Map(), je = /* @__PURE__ */ new Map(), ye = (e) => new Promise((t) => setTimeout(t, e)), Qe = async (e, t = 0) => {
  const s = new URL(e).hostname;
  if (je.get(s))
    return {
      ogImage: "",
      ogTitle: "",
      ogDescription: ""
    };
  if (O1.has(e))
    return O1.get(e);
  if (b1.has(e))
    return b1.get(e);
  const o = (async () => {
    try {
      await ye(100);
      const i = await fetch(`${Ot}?url=${encodeURIComponent(e)}`, {
        method: "GET"
      });
      if (!i.ok) {
        if (i.status === 429 && t < 5) {
          const y = Math.pow(2, t) * 1e3;
          return await ye(y), Qe(e, t + 1);
        }
        throw je.set(s, !0), new Error(`Failed to fetch Open Graph data: ${i.statusText}`);
      }
      const r = await i.json(), u = new DOMParser().parseFromString(r.contents, "text/html"), d = u.querySelector('meta[property="og:image"]'), l = u.querySelector('meta[property="og:title"]'), m = u.querySelector('meta[property="og:description"]'), f = u.querySelector("title"), C = d && d.getAttribute("content") || "", c = l ? l.getAttribute("content") || "" : f ? f.innerText : "", x = m && m.getAttribute("content") || "", p = {
        ogImage: C,
        ogTitle: c,
        ogDescription: x
      };
      return O1.set(e, p), p;
    } catch (i) {
      return console.error("Error fetching Open Graph data", i), {
        ogImage: "",
        ogTitle: "",
        ogDescription: ""
      };
    } finally {
      b1.delete(e);
    }
  })();
  return b1.set(e, o), o;
}, J1 = [];
let H1 = 0;
const Bt = 5, zt = (e) => {
  J1.push(e), Xe();
}, Xe = async () => {
  if (H1 >= Bt || J1.length === 0)
    return;
  H1++;
  const e = J1.shift();
  e && await e(), H1--, Xe();
}, Zt = ({
  src: e,
  alt: t,
  style: s
}) => /* @__PURE__ */ n.jsx("img", {
  src: e,
  alt: t,
  style: {
    display: "block",
    height: "auto",
    maxHeight: "150px",
    width: "auto",
    maxWidth: "100%",
    objectFit: "cover",
    border: "1px solid grey",
    borderRadius: "10px",
    margin: "10px 0",
    ...s
  }
}), Yt = ({
  url: e
}) => {
  const t = V.useRef(null);
  return V.useEffect(() => {
    if (t.current)
      if (M1.isSupported() && e.endsWith(".m3u8")) {
        const s = new M1();
        s.loadSource(e), s.attachMedia(t.current), s.on(M1.Events.MANIFEST_PARSED, () => {
          t.current.play();
        });
      } else
        t.current.src = e, t.current.addEventListener("loadedmetadata", () => {
          t.current.play();
        });
  }, [e]), /* @__PURE__ */ n.jsx("video", {
    ref: t,
    controls: !0,
    muted: !0,
    style: {
      width: "auto",
      maxWidth: "100%",
      maxHeight: "400px",
      borderRadius: "10px",
      margin: "10px 0",
      objectFit: "contain"
    }
  });
}, Gt = (e) => /\.(jpeg|jpg|gif|png|webp|bmp|svg)$/.test(e) || e.startsWith("https://imagedelivery.net"), Jt = (e, t, s) => {
  const [o, i] = V.useState([]), r = V.useCallback(async (a) => (await Promise.all(a.map(async (d) => {
    if (d.url) {
      const l = d.url;
      return Gt(l) ? /* @__PURE__ */ n.jsx(Zt, {
        src: l,
        alt: "Embedded image"
      }, l) : l.endsWith(".m3u8") || l.endsWith(".mp4") ? /* @__PURE__ */ n.jsx(Yt, {
        url: l
      }, l) : new Promise((m) => {
        zt(async () => {
          const {
            ogImage: f,
            ogTitle: C
          } = await Qe(l), c = new URL(l).hostname.replace("www.", "");
          m(/* @__PURE__ */ n.jsxs(Wt, {
            href: l,
            target: "_blank",
            rel: "noreferrer",
            children: [f && /* @__PURE__ */ n.jsx("img", {
              src: f,
              alt: C,
              style: {
                width: "50px",
                height: "50px",
                objectFit: "cover",
                borderRadius: "5px"
              }
            }), /* @__PURE__ */ n.jsxs("div", {
              style: {
                display: "flex",
                flexDirection: "column"
              },
              children: [/* @__PURE__ */ n.jsx("p", {
                style: {
                  margin: 0
                },
                children: C || l
              }), /* @__PURE__ */ n.jsx("p", {
                style: {
                  margin: 0,
                  color: "grey",
                  fontSize: "12px"
                },
                children: c
              })]
            })]
          }, l));
        });
      });
    } else if (d.cast_id)
      return /* @__PURE__ */ n.jsx("div", {
        style: {
          maxWidth: "85%"
        },
        children: /* @__PURE__ */ n.jsx(qt, {
          type: "hash",
          identifier: d.cast_id.hash,
          viewerFid: s,
          allowReactions: t,
          renderEmbeds: !1
        }, d.cast_id.fid)
      }, `cast-${d.cast_id.hash}`);
    return null;
  }))).filter((d) => d !== null), [t, s]);
  return V.useEffect(() => {
    r(e).then(i);
  }, [e, r]), o;
}, Kt = ({
  onClick: e
}) => /* @__PURE__ */ n.jsx("svg", {
  width: "18",
  height: "17",
  viewBox: "0 0 18 17",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  onClick: (t) => e ? e(t) : void 0,
  style: {
    cursor: "pointer"
  },
  children: /* @__PURE__ */ n.jsx("path", {
    d: "M6.22857 12.6828H7.01285C7.01285 12.2496 6.66172 11.8985 6.22857 11.8985V12.6828ZM6.22857 15.8199H5.4443C5.4443 16.1137 5.60848 16.3828 5.8697 16.5172C6.13092 16.6517 6.44537 16.6288 6.68443 16.4581L6.22857 15.8199ZM10.6205 12.6828V11.8985C10.4571 11.8985 10.2977 11.9496 10.1647 12.0446L10.6205 12.6828ZM1.78428 3.27146C1.78428 2.54955 2.3695 1.96433 3.09141 1.96433V0.395777C1.50321 0.395777 0.215723 1.68327 0.215723 3.27146H1.78428ZM1.78428 10.5914V3.27146H0.215723V10.5914H1.78428ZM3.0914 11.8985C2.3695 11.8985 1.78428 11.3133 1.78428 10.5914H0.215723C0.215723 12.1796 1.50321 13.4671 3.0914 13.4671V11.8985ZM6.22857 11.8985H3.0914V13.4671H6.22857V11.8985ZM7.01285 15.8199V12.6828H5.4443V15.8199H7.01285ZM10.1647 12.0446L5.77272 15.1817L6.68443 16.4581L11.0764 13.321L10.1647 12.0446ZM14.5941 11.8985H10.6205V13.4671H14.5941V11.8985ZM15.9013 10.5914C15.9013 11.3133 15.316 11.8985 14.5941 11.8985V13.4671C16.1823 13.4671 17.4698 12.1796 17.4698 10.5914H15.9013ZM15.9013 3.27146V10.5914H17.4698V3.27146H15.9013ZM14.5941 1.96433C15.316 1.96433 15.9013 2.54955 15.9013 3.27146H17.4698C17.4698 1.68326 16.1823 0.395777 14.5941 0.395777V1.96433ZM3.09141 1.96433H14.5941V0.395777H3.09141V1.96433Z",
    fill: "#A0A3AD"
  })
}), Qt = ({
  fill: e,
  onClick: t
}) => /* @__PURE__ */ n.jsxs("svg", {
  width: "16",
  height: "15",
  viewBox: "0 0 16 15",
  fill: e || "none",
  xmlns: "http://www.w3.org/2000/svg",
  onClick: (s) => t ? t(s) : void 0,
  style: {
    cursor: "pointer"
  },
  children: [/* @__PURE__ */ n.jsx("path", {
    d: "M14.9245 14.2506C14.7258 14.2506 14.5352 14.1716 14.3946 14.0311C14.2541 13.8905 14.1751 13.6999 14.1751 13.5011V10.5034H11.1774C10.9787 10.5034 10.7881 10.4245 10.6475 10.2839C10.507 10.1434 10.428 9.95279 10.428 9.75403C10.428 9.55527 10.507 9.36465 10.6475 9.22411C10.7881 9.08357 10.9787 9.00461 11.1774 9.00461H14.9245C15.1233 9.00461 15.3139 9.08357 15.4545 9.22411C15.595 9.36465 15.674 9.55527 15.674 9.75403V13.5011C15.674 13.6999 15.595 13.8905 15.4545 14.0311C15.3139 14.1716 15.1233 14.2506 14.9245 14.2506ZM5.18207 6.00693H1.43497C1.23621 6.00693 1.04559 5.92797 0.905047 5.78743C0.764503 5.64688 0.685547 5.45627 0.685547 5.25751V1.51041C0.685547 1.31165 0.764503 1.12103 0.905047 0.980486C1.04559 0.839943 1.23621 0.760986 1.43497 0.760986C1.63373 0.760986 1.82434 0.839943 1.96489 0.980486C2.10543 1.12103 2.18439 1.31165 2.18439 1.51041V4.50809H5.18207C5.38083 4.50809 5.57145 4.58704 5.71199 4.72759C5.85253 4.86813 5.93149 5.05875 5.93149 5.25751C5.93149 5.45627 5.85253 5.64688 5.71199 5.78743C5.57145 5.92797 5.38083 6.00693 5.18207 6.00693Z",
    fill: e || "#A0A3AD"
  }), /* @__PURE__ */ n.jsx("path", {
    d: "M8.17934 15C6.33556 15.0003 4.55637 14.3209 3.18205 13.0918C1.80774 11.8627 0.934766 10.17 0.730106 8.33765C0.719281 8.23923 0.727945 8.13965 0.755606 8.04458C0.783266 7.94951 0.82938 7.86083 0.891315 7.78358C1.0164 7.62758 1.19833 7.52765 1.39709 7.50579C1.59585 7.48393 1.79515 7.54191 1.95116 7.667C2.10716 7.79208 2.20708 7.97401 2.22895 8.17277C2.37708 9.52376 2.97983 10.7843 3.93838 11.7478C4.89692 12.7113 6.15438 13.3205 7.50459 13.4755C8.85479 13.6306 10.2176 13.3223 11.3696 12.6012C12.5216 11.8801 13.3944 10.789 13.845 9.50674C13.8731 9.40817 13.9211 9.31642 13.986 9.23714C14.051 9.15787 14.1316 9.09277 14.2227 9.04586C14.3138 8.99895 14.4136 8.97123 14.5159 8.96442C14.6181 8.95761 14.7207 8.97185 14.8173 9.00625C14.9138 9.04066 15.0023 9.09451 15.0772 9.16446C15.1521 9.23442 15.2118 9.31899 15.2528 9.41296C15.2937 9.50693 15.3149 9.6083 15.3151 9.71079C15.3153 9.81329 15.2944 9.91473 15.2539 10.0089C14.7359 11.4705 13.7774 12.7355 12.5103 13.6294C11.2432 14.5233 9.73002 15.0022 8.17934 15ZM14.8792 7.50579C14.6939 7.50691 14.5148 7.43937 14.3764 7.3162C14.238 7.19302 14.1501 7.02295 14.1297 6.8388C13.975 5.49429 13.3699 4.24169 12.4128 3.28476C11.4558 2.32782 10.2031 1.72285 8.85858 1.56824C7.51405 1.41364 6.15676 1.7185 5.0075 2.43324C3.85823 3.14798 2.9846 4.23054 2.52872 5.50484C2.50062 5.6034 2.4526 5.69516 2.38763 5.77443C2.32266 5.85371 2.24212 5.91881 2.15099 5.96572C2.05987 6.01263 1.96008 6.04035 1.85781 6.04716C1.75555 6.05397 1.65297 6.03973 1.55642 6.00532C1.45988 5.97092 1.37141 5.91707 1.29651 5.84712C1.2216 5.77716 1.16183 5.69258 1.12091 5.59861C1.07999 5.50464 1.05878 5.40328 1.05859 5.30078C1.0584 5.19829 1.07923 5.09685 1.1198 5.00273C1.68406 3.40202 2.77497 2.04025 4.21397 1.14029C5.65297 0.240337 7.35485 -0.144505 9.04106 0.0487522C10.7273 0.242009 12.2979 1.00192 13.496 2.20411C14.6941 3.4063 15.4486 4.97958 15.6361 6.66644C15.6571 6.86306 15.5997 7.06004 15.4761 7.21445C15.3526 7.36887 15.173 7.4682 14.9766 7.4908L14.8792 7.50579Z",
    fill: e || "#A0A3AD"
  })]
}), Xt = ({
  fill: e,
  onClick: t
}) => /* @__PURE__ */ n.jsxs("svg", {
  width: "17",
  height: "15",
  viewBox: "0 0 17 15",
  fill: e ? "red" : "none",
  xmlns: "http://www.w3.org/2000/svg",
  onClick: (s) => t ? t(s) : void 0,
  style: {
    cursor: "pointer"
  },
  children: [/* @__PURE__ */ n.jsx("path", {
    d: "M13.8719 2.25042L14.2947 1.58984L13.8719 2.25042C15.2467 3.13022 15.7676 5.22206 14.8856 7.61317C14.0436 9.89583 11.9555 12.2374 8.51573 13.3648C5.07606 12.2374 2.98799 9.89584 2.14596 7.61316C1.26392 5.22203 1.7848 3.13022 3.15951 2.25042L3.15951 2.25042C4.58542 1.33783 5.64997 1.52983 6.39945 1.92503C7.21272 2.35388 7.73108 3.07114 7.86143 3.26838L8.51573 4.25838L9.17002 3.26838C9.30038 3.07113 9.81873 2.35388 10.632 1.92503C11.3815 1.52983 12.446 1.33783 13.8719 2.25042Z",
    stroke: e || "#A0A3AD",
    strokeWidth: "1.56855"
  }), /* @__PURE__ */ n.jsx("mask", {
    id: "path-2-inside-1_187_1051",
    fill: "white",
    children: /* @__PURE__ */ n.jsx("path", {
      d: "M8.58444 13.9397C7.76139 13.6864 7.05334 13.3937 6.29976 12.9825C5.95633 12.7951 5.62285 12.5895 5.3003 12.3681C5.14416 12.2609 4.99108 12.1493 4.84063 12.0343C4.78132 11.9889 4.88038 12.0652 4.82156 12.0195C4.80388 12.0058 4.7863 11.9919 4.76869 11.9781C4.73364 11.9505 4.69884 11.9225 4.66413 11.8945C4.58948 11.8342 4.51582 11.7727 4.4428 11.7105C3.90499 11.252 3.41381 10.7389 2.98016 10.1809C3.01856 10.2303 2.97651 10.176 2.97064 10.1683C2.9583 10.1522 2.94604 10.1359 2.9338 10.1197C2.90939 10.0874 2.88527 10.0548 2.86126 10.0221C2.80961 9.95189 2.75908 9.8808 2.70932 9.8092C2.61492 9.67345 2.52441 9.535 2.4371 9.39457C2.2673 9.12145 2.11213 8.83926 1.97147 8.55006C1.93849 8.48228 1.90657 8.41396 1.87531 8.34537C1.85978 8.3113 1.84459 8.27709 1.82948 8.24284C1.82195 8.22575 1.81456 8.20864 1.80714 8.19152C1.80167 8.17889 1.76856 8.10061 1.79664 8.16736C1.73728 8.0262 1.68205 7.88329 1.63029 7.73916C1.52982 7.45942 1.44498 7.17407 1.37575 6.88502C1.34218 6.74492 1.31301 6.60379 1.28746 6.46202C1.2754 6.39502 1.2645 6.32782 1.2543 6.26052C1.24925 6.22713 1.24453 6.19369 1.23994 6.16023C1.25136 6.24358 1.23495 6.11772 1.23231 6.09532C1.20042 5.82444 1.18535 5.55158 1.18721 5.27885C1.18811 5.14691 1.19374 5.01501 1.20334 4.88343C1.20788 4.82128 1.21368 4.75922 1.22023 4.69726C1.2235 4.66651 1.22713 4.6358 1.23089 4.60511C1.23278 4.58982 1.23477 4.57457 1.23678 4.5593C1.22562 4.64425 1.23563 4.56956 1.23823 4.55163C1.27428 4.30419 1.32861 4.05943 1.40097 3.82006C1.43614 3.70375 1.47635 3.58899 1.52072 3.47587C1.53125 3.449 1.54219 3.42229 1.55326 3.39564C1.51941 3.47716 1.55889 3.38334 1.56717 3.36464C1.59226 3.30803 1.61886 3.25209 1.64643 3.19664C1.75101 2.98624 1.87476 2.78549 2.01534 2.59724C2.02371 2.58603 2.03221 2.57492 2.04071 2.5638C1.99275 2.62656 2.02446 2.58496 2.03493 2.57169C2.05223 2.54974 2.06993 2.52808 2.08773 2.50653C2.12631 2.45984 2.16629 2.41434 2.20713 2.36962C2.28587 2.28341 2.36926 2.20151 2.456 2.12339C2.49787 2.08568 2.54084 2.0492 2.58443 2.01352C2.60631 1.99561 2.62852 1.97808 2.65084 1.9607C2.58164 2.01466 2.67191 1.94557 2.68962 1.93265C2.83338 1.8277 2.98587 1.7348 3.14013 1.6463C3.54393 1.41459 3.92437 1.25567 4.3831 1.15072C4.47716 1.12921 4.57219 1.11204 4.6676 1.09771C4.74483 1.08611 4.61527 1.10399 4.69217 1.09444C4.71513 1.0916 4.73817 1.08917 4.76119 1.08677C4.81403 1.08124 4.86704 1.07739 4.92007 1.07423C5.10084 1.06348 5.28237 1.06851 5.46252 1.0864C5.48336 1.08846 5.50417 1.09092 5.52495 1.09338C5.59508 1.10167 5.46943 1.08526 5.53899 1.09521C5.57965 1.10103 5.62016 1.10795 5.66059 1.11524C5.74647 1.13073 5.83159 1.1503 5.91604 1.17227C6.0728 1.21311 6.22621 1.26621 6.37595 1.32786C6.31111 1.30115 6.41877 1.34702 6.43584 1.35477C6.46987 1.3702 6.50356 1.38639 6.53714 1.4028C6.60865 1.43779 6.67886 1.47543 6.74817 1.51462C6.87662 1.58723 7.00062 1.66753 7.12092 1.7529C7.14882 1.77271 7.17637 1.79297 7.20383 1.81338C7.21739 1.82348 7.23087 1.8337 7.24432 1.84396C7.29158 1.87992 7.1988 1.80839 7.24529 1.84482C7.30109 1.88853 7.35568 1.93375 7.40949 1.97982C7.50652 2.06288 7.59994 2.15008 7.69053 2.24011C7.84008 2.38879 7.98044 2.54677 8.11149 2.71194C8.12572 2.72987 8.17533 2.79414 8.11485 2.7157C8.12636 2.73065 8.13776 2.74568 8.1491 2.76075C8.1694 2.78769 8.18934 2.81491 8.20905 2.8423C8.23825 2.88286 8.2667 2.92398 8.29429 2.96566C8.40374 3.131 8.6286 3.131 8.73805 2.96566C8.76707 2.92181 8.79712 2.87866 8.82785 2.83598C8.84783 2.80823 8.86819 2.78075 8.88875 2.75344C8.90242 2.73529 8.94928 2.67474 8.88935 2.75214C8.90409 2.73312 8.919 2.71424 8.93399 2.69542C9.0671 2.52834 9.21071 2.36968 9.36261 2.21954C9.45411 2.12907 9.5493 2.04234 9.64732 1.95897C9.69737 1.91641 9.74847 1.87507 9.80017 1.83456C9.7388 1.88263 9.82694 1.81457 9.84213 1.80327C9.87427 1.77937 9.90683 1.75606 9.9396 1.73301C10.0608 1.64773 10.1866 1.56915 10.3159 1.49682C10.3802 1.46084 10.4457 1.42703 10.5119 1.39467C10.5455 1.37823 10.5795 1.36251 10.6136 1.34704C10.6308 1.33926 10.6481 1.33171 10.6654 1.32417C10.6915 1.31296 10.6885 1.3142 10.6564 1.3279C10.6681 1.32303 10.6798 1.31822 10.6916 1.31349C10.8423 1.25235 10.9973 1.20246 11.1548 1.16239C11.2332 1.14248 11.3123 1.12599 11.3918 1.11166C11.4323 1.10435 11.473 1.09815 11.5138 1.09231C11.4471 1.10186 11.5171 1.09226 11.5281 1.09095C11.5559 1.08767 11.5838 1.08499 11.6117 1.0824C11.793 1.06553 11.9757 1.06513 12.1573 1.0771C12.203 1.08011 12.2485 1.08444 12.294 1.08921C12.3172 1.09163 12.3402 1.0945 12.3633 1.09734C12.3036 1.08994 12.3813 1.10022 12.3881 1.10123C12.4921 1.11685 12.5952 1.13785 12.6975 1.16193C13.1586 1.27048 13.5378 1.44037 13.9464 1.67761C14.0904 1.76129 14.2314 1.8507 14.3656 1.9495C14.377 1.95791 14.3884 1.96648 14.3997 1.97503C14.4419 2.00688 14.351 1.93696 14.3926 1.96945C14.4149 1.98683 14.4369 2.00462 14.4588 2.02252C14.5059 2.06115 14.5519 2.1013 14.5971 2.14228C14.6835 2.22073 14.7656 2.30388 14.8439 2.39035C14.8815 2.43176 14.9178 2.47427 14.9534 2.51736C14.9712 2.53891 14.9887 2.56077 15.006 2.58274C15.0384 2.62388 14.9682 2.53335 15 2.575C15.0114 2.58983 15.0225 2.60479 15.0336 2.61978C15.1736 2.80871 15.2946 3.01142 15.3985 3.22224C15.4238 3.27356 15.4477 3.32553 15.4709 3.37782C15.4767 3.39096 15.4824 3.40414 15.4881 3.41734C15.5093 3.46656 15.464 3.35945 15.4845 3.40902C15.4975 3.44013 15.5099 3.47148 15.5221 3.50289C15.5661 3.61632 15.6049 3.73175 15.6398 3.84833C15.7115 4.08832 15.763 4.33416 15.7984 4.58204C15.8064 4.63806 15.7901 4.51842 15.7975 4.57463C15.7995 4.58992 15.8014 4.60521 15.8033 4.6205C15.807 4.6512 15.8104 4.68197 15.8137 4.71273C15.8208 4.77994 15.8264 4.84732 15.8312 4.91474C15.8404 5.04658 15.8447 5.17876 15.8453 5.31093C15.8464 5.58418 15.8287 5.85737 15.7961 6.12862C15.7941 6.14529 15.7919 6.16192 15.7898 6.17856C15.7822 6.23827 15.7983 6.11722 15.7901 6.17697C15.7855 6.21044 15.7806 6.24389 15.7755 6.27731C15.7644 6.35031 15.752 6.4231 15.7388 6.49573C15.7129 6.63769 15.6823 6.7788 15.6484 6.91905C15.5784 7.20843 15.4909 7.49348 15.3897 7.77343C15.3648 7.84232 15.3388 7.9108 15.3121 7.97902C15.2987 8.01338 15.2849 8.0476 15.2711 8.08179C15.2641 8.09896 15.257 8.11604 15.25 8.13315C15.2452 8.14458 15.2405 8.15598 15.2357 8.16738C15.2496 8.13453 15.251 8.13117 15.2399 8.15734C15.1786 8.30137 15.1131 8.44355 15.0443 8.58416C14.9028 8.87351 14.7451 9.15483 14.5745 9.42799C14.4868 9.56844 14.395 9.70634 14.3002 9.84213C14.254 9.90821 14.2069 9.97355 14.1592 10.0385C14.1351 10.0711 14.1108 10.1036 14.0864 10.136C14.0741 10.1522 14.0617 10.1684 14.0494 10.1845C14.0319 10.2073 14.0329 10.2061 14.0522 10.181C14.0439 10.1918 14.0355 10.2025 14.0271 10.2132C13.8135 10.4869 13.5849 10.7488 13.3445 10.9992C13.0948 11.2594 12.8309 11.5058 12.556 11.7391C12.4884 11.7965 12.4199 11.8528 12.3509 11.9086C12.3162 11.9366 12.2812 11.9643 12.2461 11.992C12.2285 12.0058 12.2108 12.0196 12.1931 12.0333C12.2377 11.9987 12.1636 12.0558 12.156 12.0616C12.005 12.1765 11.8508 12.2871 11.6942 12.3941C11.3706 12.6153 11.0348 12.8185 10.6904 13.0055C10.5122 13.1023 10.3313 13.1939 10.1484 13.2815C10.059 13.3243 9.96904 13.3658 9.87868 13.4064C9.8331 13.4269 9.78733 13.4469 9.74151 13.4669C9.7185 13.4769 9.69541 13.4868 9.67233 13.4966C9.6442 13.5086 9.64146 13.5098 9.66417 13.5002C9.64864 13.5067 9.63313 13.5132 9.61758 13.5196C9.2349 13.6788 8.84398 13.8179 8.44786 13.9398C8.31409 13.9809 8.2304 14.1178 8.26839 14.2559C8.3047 14.3878 8.45064 14.4765 8.5845 14.4353C10.1534 13.9525 11.626 13.1871 12.8857 12.131C13.9425 11.245 14.8349 10.1456 15.4543 8.91165C15.9942 7.83608 16.3357 6.64744 16.3583 5.43967C16.3774 4.41628 16.149 3.35704 15.5742 2.49789C15.2929 2.07732 14.9371 1.71816 14.5205 1.43134C13.7956 0.932295 12.9294 0.592822 12.045 0.557214C10.7717 0.505949 9.60126 1.17157 8.7697 2.09685C8.59779 2.28813 8.43634 2.49172 8.29431 2.70629C8.44223 2.70629 8.59015 2.70629 8.73807 2.70629C8.22434 1.93023 7.48255 1.25338 6.6264 0.880424C5.9305 0.577258 5.18555 0.490164 4.43721 0.617179C3.98766 0.69348 3.55546 0.856508 3.14849 1.05915C2.81945 1.22296 2.50653 1.41842 2.22115 1.65017C1.44904 2.27713 0.973026 3.20995 0.783732 4.17519C0.56509 5.29004 0.68737 6.45381 1.02051 7.53339C1.41084 8.79827 2.09407 9.95916 2.95641 10.9589C3.94757 12.1081 5.19614 13.0306 6.5582 13.6933C7.16712 13.9896 7.8008 14.2362 8.44788 14.4353C8.58174 14.4765 8.72768 14.3878 8.76399 14.2559C8.80189 14.1177 8.71821 13.9809 8.58444 13.9397Z"
    })
  }), /* @__PURE__ */ n.jsx("path", {
    d: "M8.58444 13.9397C7.76139 13.6864 7.05334 13.3937 6.29976 12.9825C5.95633 12.7951 5.62285 12.5895 5.3003 12.3681C5.14416 12.2609 4.99108 12.1493 4.84063 12.0343C4.78132 11.9889 4.88038 12.0652 4.82156 12.0195C4.80388 12.0058 4.7863 11.9919 4.76869 11.9781C4.73364 11.9505 4.69884 11.9225 4.66413 11.8945C4.58948 11.8342 4.51582 11.7727 4.4428 11.7105C3.90499 11.252 3.41381 10.7389 2.98016 10.1809C3.01856 10.2303 2.97651 10.176 2.97064 10.1683C2.9583 10.1522 2.94604 10.1359 2.9338 10.1197C2.90939 10.0874 2.88527 10.0548 2.86126 10.0221C2.80961 9.95189 2.75908 9.8808 2.70932 9.8092C2.61492 9.67345 2.52441 9.535 2.4371 9.39457C2.2673 9.12145 2.11213 8.83926 1.97147 8.55006C1.93849 8.48228 1.90657 8.41396 1.87531 8.34537C1.85978 8.3113 1.84459 8.27709 1.82948 8.24284C1.82195 8.22575 1.81456 8.20864 1.80714 8.19152C1.80167 8.17889 1.76856 8.10061 1.79664 8.16736C1.73728 8.0262 1.68205 7.88329 1.63029 7.73916C1.52982 7.45942 1.44498 7.17407 1.37575 6.88502C1.34218 6.74492 1.31301 6.60379 1.28746 6.46202C1.2754 6.39502 1.2645 6.32782 1.2543 6.26052C1.24925 6.22713 1.24453 6.19369 1.23994 6.16023C1.25136 6.24358 1.23495 6.11772 1.23231 6.09532C1.20042 5.82444 1.18535 5.55158 1.18721 5.27885C1.18811 5.14691 1.19374 5.01501 1.20334 4.88343C1.20788 4.82128 1.21368 4.75922 1.22023 4.69726C1.2235 4.66651 1.22713 4.6358 1.23089 4.60511C1.23278 4.58982 1.23477 4.57457 1.23678 4.5593C1.22562 4.64425 1.23563 4.56956 1.23823 4.55163C1.27428 4.30419 1.32861 4.05943 1.40097 3.82006C1.43614 3.70375 1.47635 3.58899 1.52072 3.47587C1.53125 3.449 1.54219 3.42229 1.55326 3.39564C1.51941 3.47716 1.55889 3.38334 1.56717 3.36464C1.59226 3.30803 1.61886 3.25209 1.64643 3.19664C1.75101 2.98624 1.87476 2.78549 2.01534 2.59724C2.02371 2.58603 2.03221 2.57492 2.04071 2.5638C1.99275 2.62656 2.02446 2.58496 2.03493 2.57169C2.05223 2.54974 2.06993 2.52808 2.08773 2.50653C2.12631 2.45984 2.16629 2.41434 2.20713 2.36962C2.28587 2.28341 2.36926 2.20151 2.456 2.12339C2.49787 2.08568 2.54084 2.0492 2.58443 2.01352C2.60631 1.99561 2.62852 1.97808 2.65084 1.9607C2.58164 2.01466 2.67191 1.94557 2.68962 1.93265C2.83338 1.8277 2.98587 1.7348 3.14013 1.6463C3.54393 1.41459 3.92437 1.25567 4.3831 1.15072C4.47716 1.12921 4.57219 1.11204 4.6676 1.09771C4.74483 1.08611 4.61527 1.10399 4.69217 1.09444C4.71513 1.0916 4.73817 1.08917 4.76119 1.08677C4.81403 1.08124 4.86704 1.07739 4.92007 1.07423C5.10084 1.06348 5.28237 1.06851 5.46252 1.0864C5.48336 1.08846 5.50417 1.09092 5.52495 1.09338C5.59508 1.10167 5.46943 1.08526 5.53899 1.09521C5.57965 1.10103 5.62016 1.10795 5.66059 1.11524C5.74647 1.13073 5.83159 1.1503 5.91604 1.17227C6.0728 1.21311 6.22621 1.26621 6.37595 1.32786C6.31111 1.30115 6.41877 1.34702 6.43584 1.35477C6.46987 1.3702 6.50356 1.38639 6.53714 1.4028C6.60865 1.43779 6.67886 1.47543 6.74817 1.51462C6.87662 1.58723 7.00062 1.66753 7.12092 1.7529C7.14882 1.77271 7.17637 1.79297 7.20383 1.81338C7.21739 1.82348 7.23087 1.8337 7.24432 1.84396C7.29158 1.87992 7.1988 1.80839 7.24529 1.84482C7.30109 1.88853 7.35568 1.93375 7.40949 1.97982C7.50652 2.06288 7.59994 2.15008 7.69053 2.24011C7.84008 2.38879 7.98044 2.54677 8.11149 2.71194C8.12572 2.72987 8.17533 2.79414 8.11485 2.7157C8.12636 2.73065 8.13776 2.74568 8.1491 2.76075C8.1694 2.78769 8.18934 2.81491 8.20905 2.8423C8.23825 2.88286 8.2667 2.92398 8.29429 2.96566C8.40374 3.131 8.6286 3.131 8.73805 2.96566C8.76707 2.92181 8.79712 2.87866 8.82785 2.83598C8.84783 2.80823 8.86819 2.78075 8.88875 2.75344C8.90242 2.73529 8.94928 2.67474 8.88935 2.75214C8.90409 2.73312 8.919 2.71424 8.93399 2.69542C9.0671 2.52834 9.21071 2.36968 9.36261 2.21954C9.45411 2.12907 9.5493 2.04234 9.64732 1.95897C9.69737 1.91641 9.74847 1.87507 9.80017 1.83456C9.7388 1.88263 9.82694 1.81457 9.84213 1.80327C9.87427 1.77937 9.90683 1.75606 9.9396 1.73301C10.0608 1.64773 10.1866 1.56915 10.3159 1.49682C10.3802 1.46084 10.4457 1.42703 10.5119 1.39467C10.5455 1.37823 10.5795 1.36251 10.6136 1.34704C10.6308 1.33926 10.6481 1.33171 10.6654 1.32417C10.6915 1.31296 10.6885 1.3142 10.6564 1.3279C10.6681 1.32303 10.6798 1.31822 10.6916 1.31349C10.8423 1.25235 10.9973 1.20246 11.1548 1.16239C11.2332 1.14248 11.3123 1.12599 11.3918 1.11166C11.4323 1.10435 11.473 1.09815 11.5138 1.09231C11.4471 1.10186 11.5171 1.09226 11.5281 1.09095C11.5559 1.08767 11.5838 1.08499 11.6117 1.0824C11.793 1.06553 11.9757 1.06513 12.1573 1.0771C12.203 1.08011 12.2485 1.08444 12.294 1.08921C12.3172 1.09163 12.3402 1.0945 12.3633 1.09734C12.3036 1.08994 12.3813 1.10022 12.3881 1.10123C12.4921 1.11685 12.5952 1.13785 12.6975 1.16193C13.1586 1.27048 13.5378 1.44037 13.9464 1.67761C14.0904 1.76129 14.2314 1.8507 14.3656 1.9495C14.377 1.95791 14.3884 1.96648 14.3997 1.97503C14.4419 2.00688 14.351 1.93696 14.3926 1.96945C14.4149 1.98683 14.4369 2.00462 14.4588 2.02252C14.5059 2.06115 14.5519 2.1013 14.5971 2.14228C14.6835 2.22073 14.7656 2.30388 14.8439 2.39035C14.8815 2.43176 14.9178 2.47427 14.9534 2.51736C14.9712 2.53891 14.9887 2.56077 15.006 2.58274C15.0384 2.62388 14.9682 2.53335 15 2.575C15.0114 2.58983 15.0225 2.60479 15.0336 2.61978C15.1736 2.80871 15.2946 3.01142 15.3985 3.22224C15.4238 3.27356 15.4477 3.32553 15.4709 3.37782C15.4767 3.39096 15.4824 3.40414 15.4881 3.41734C15.5093 3.46656 15.464 3.35945 15.4845 3.40902C15.4975 3.44013 15.5099 3.47148 15.5221 3.50289C15.5661 3.61632 15.6049 3.73175 15.6398 3.84833C15.7115 4.08832 15.763 4.33416 15.7984 4.58204C15.8064 4.63806 15.7901 4.51842 15.7975 4.57463C15.7995 4.58992 15.8014 4.60521 15.8033 4.6205C15.807 4.6512 15.8104 4.68197 15.8137 4.71273C15.8208 4.77994 15.8264 4.84732 15.8312 4.91474C15.8404 5.04658 15.8447 5.17876 15.8453 5.31093C15.8464 5.58418 15.8287 5.85737 15.7961 6.12862C15.7941 6.14529 15.7919 6.16192 15.7898 6.17856C15.7822 6.23827 15.7983 6.11722 15.7901 6.17697C15.7855 6.21044 15.7806 6.24389 15.7755 6.27731C15.7644 6.35031 15.752 6.4231 15.7388 6.49573C15.7129 6.63769 15.6823 6.7788 15.6484 6.91905C15.5784 7.20843 15.4909 7.49348 15.3897 7.77343C15.3648 7.84232 15.3388 7.9108 15.3121 7.97902C15.2987 8.01338 15.2849 8.0476 15.2711 8.08179C15.2641 8.09896 15.257 8.11604 15.25 8.13315C15.2452 8.14458 15.2405 8.15598 15.2357 8.16738C15.2496 8.13453 15.251 8.13117 15.2399 8.15734C15.1786 8.30137 15.1131 8.44355 15.0443 8.58416C14.9028 8.87351 14.7451 9.15483 14.5745 9.42799C14.4868 9.56844 14.395 9.70634 14.3002 9.84213C14.254 9.90821 14.2069 9.97355 14.1592 10.0385C14.1351 10.0711 14.1108 10.1036 14.0864 10.136C14.0741 10.1522 14.0617 10.1684 14.0494 10.1845C14.0319 10.2073 14.0329 10.2061 14.0522 10.181C14.0439 10.1918 14.0355 10.2025 14.0271 10.2132C13.8135 10.4869 13.5849 10.7488 13.3445 10.9992C13.0948 11.2594 12.8309 11.5058 12.556 11.7391C12.4884 11.7965 12.4199 11.8528 12.3509 11.9086C12.3162 11.9366 12.2812 11.9643 12.2461 11.992C12.2285 12.0058 12.2108 12.0196 12.1931 12.0333C12.2377 11.9987 12.1636 12.0558 12.156 12.0616C12.005 12.1765 11.8508 12.2871 11.6942 12.3941C11.3706 12.6153 11.0348 12.8185 10.6904 13.0055C10.5122 13.1023 10.3313 13.1939 10.1484 13.2815C10.059 13.3243 9.96904 13.3658 9.87868 13.4064C9.8331 13.4269 9.78733 13.4469 9.74151 13.4669C9.7185 13.4769 9.69541 13.4868 9.67233 13.4966C9.6442 13.5086 9.64146 13.5098 9.66417 13.5002C9.64864 13.5067 9.63313 13.5132 9.61758 13.5196C9.2349 13.6788 8.84398 13.8179 8.44786 13.9398C8.31409 13.9809 8.2304 14.1178 8.26839 14.2559C8.3047 14.3878 8.45064 14.4765 8.5845 14.4353C10.1534 13.9525 11.626 13.1871 12.8857 12.131C13.9425 11.245 14.8349 10.1456 15.4543 8.91165C15.9942 7.83608 16.3357 6.64744 16.3583 5.43967C16.3774 4.41628 16.149 3.35704 15.5742 2.49789C15.2929 2.07732 14.9371 1.71816 14.5205 1.43134C13.7956 0.932295 12.9294 0.592822 12.045 0.557214C10.7717 0.505949 9.60126 1.17157 8.7697 2.09685C8.59779 2.28813 8.43634 2.49172 8.29431 2.70629C8.44223 2.70629 8.59015 2.70629 8.73807 2.70629C8.22434 1.93023 7.48255 1.25338 6.6264 0.880424C5.9305 0.577258 5.18555 0.490164 4.43721 0.617179C3.98766 0.69348 3.55546 0.856508 3.14849 1.05915C2.81945 1.22296 2.50653 1.41842 2.22115 1.65017C1.44904 2.27713 0.973026 3.20995 0.783732 4.17519C0.56509 5.29004 0.68737 6.45381 1.02051 7.53339C1.41084 8.79827 2.09407 9.95916 2.95641 10.9589C3.94757 12.1081 5.19614 13.0306 6.5582 13.6933C7.16712 13.9896 7.8008 14.2362 8.44788 14.4353C8.58174 14.4765 8.72768 14.3878 8.76399 14.2559C8.80189 14.1177 8.71821 13.9809 8.58444 13.9397Z",
    stroke: e || "#A0A3AD",
    strokeWidth: "3.13711",
    mask: "url(#path-2-inside-1_187_1051)"
  })]
});
function et({
  onClick: e,
  size: t = 24
}) {
  return /* @__PURE__ */ n.jsx("svg", {
    "aria-hidden": "true",
    focusable: "false",
    role: "img",
    viewBox: "0 0 16 16",
    width: t,
    height: t,
    fill: "#4A4A4A",
    style: {
      display: "inline-block",
      userSelect: "none",
      verticalAlign: "text-bottom",
      overflow: "visible",
      cursor: "pointer",
      padding: "4px"
    },
    onClick: e,
    children: /* @__PURE__ */ n.jsx("path", {
      d: "M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z"
    })
  });
}
const en = () => K, tn = /* @__PURE__ */ g(en())({
  classes: ["rect7e5"]
}), nn = () => K, sn = /* @__PURE__ */ g(nn())({
  classes: ["p13mqntt"]
}), rn = () => K, on = /* @__PURE__ */ g(rn())({
  classes: ["p1r47dz2"]
}), an = () => K, cn = /* @__PURE__ */ g(an())({
  classes: ["cnojpqv"]
}), ln = ({
  hash: e,
  reactions: t,
  onComment: s,
  onRecast: o,
  onLike: i,
  isLiked: r
}) => {
  const {
    client_id: a,
    user: u,
    isAuthenticated: d
  } = o1(), [l, m] = V.useState(!1), [f, C] = V.useState({
    top: 0,
    left: 0
  }), [c, x] = V.useState(null), [p, y] = V.useState(r), [v, S] = V.useState(!1), E = V.useRef(null), R = V.useRef({
    comment: null,
    recast: null,
    like: null
  });
  H(() => {
    y(t.likes.some((w) => w.fid === (u == null ? void 0 : u.fid))), S(t.recasts.some((w) => w.fid === (u == null ? void 0 : u.fid)));
  }, [t, u]), H(() => {
    const w = localStorage.getItem(w1.NEYNAR_AUTHENTICATED_USER);
    if (w)
      try {
        x(JSON.parse(w).signer_uuid);
      } catch (I) {
        console.error("Error parsing JSON from local storage:", I), x(null);
      }
    else
      console.warn("No NEYNAR_AUTHENTICATED_USER found in local storage.");
  }, [d]), H(() => {
    (c || d) && l && m(!1);
  }, [c, d, l]);
  const h = async (w, I) => {
    if (c)
      switch (I) {
        case "comment":
          if (s)
            s();
          else
            throw new Error("No comment handler function provided");
          break;
        case "recast":
          if (o)
            S(o());
          else
            throw new Error("No recast handler function provided");
          break;
        case "like":
          if (i)
            y(i());
          else
            throw new Error("No like handler function provided");
          break;
      }
    const j = R.current[I];
    if (j) {
      const T = j.getBoundingClientRect(), A = E.current;
      if (A) {
        const F = A.getBoundingClientRect();
        C({
          top: T.top - F.height - 10,
          left: T.left + T.width / 2 - F.width / 2
        });
      }
    }
  };
  return /* @__PURE__ */ n.jsxs(tn, {
    children: [l && /* @__PURE__ */ n.jsxs(sn, {
      ref: E,
      style: {
        top: f.top,
        left: f.left
      },
      children: [/* @__PURE__ */ n.jsx(on, {
        children: /* @__PURE__ */ n.jsx(Vt, {
          variant: c1.NEYNAR
        })
      }), /* @__PURE__ */ n.jsx(cn, {
        children: /* @__PURE__ */ n.jsx(et, {
          onClick: () => m(!1),
          size: 16
        })
      })]
    }), /* @__PURE__ */ n.jsx(K, {
      style: {
        display: "flex",
        justifyContent: "space-between",
        alignItems: "center"
      },
      children: /* @__PURE__ */ n.jsxs(K, {
        spacingVertical: "15px",
        style: {
          display: "flex",
          gap: "10px"
        },
        children: [/* @__PURE__ */ n.jsx("div", {
          ref: (w) => R.current.comment = w,
          children: /* @__PURE__ */ n.jsx(Kt, {
            onClick: (w) => h(w, "comment")
          })
        }), /* @__PURE__ */ n.jsx("div", {
          ref: (w) => R.current.recast = w,
          children: /* @__PURE__ */ n.jsx(Qt, {
            fill: v ? "green" : void 0,
            onClick: (w) => h(w, "recast")
          })
        }), /* @__PURE__ */ n.jsx("div", {
          ref: (w) => R.current.like = w,
          children: /* @__PURE__ */ n.jsx(Xt, {
            fill: p ? "red" : void 0,
            onClick: (w) => h(w, "like")
          })
        })]
      })
    })]
  });
};
function un({
  onClick: e
}) {
  return /* @__PURE__ */ n.jsx("svg", {
    style: {
      cursor: "pointer"
    },
    width: "16",
    height: "15",
    viewBox: "0 0 16 15",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg",
    onClick: (t) => e ? e(t) : void 0,
    children: /* @__PURE__ */ n.jsx("path", {
      d: "M15.2003 7.49063C14.7504 7.49063 14.4504 7.79057 14.4504 8.24048V12.7396C14.4504 13.1895 14.1505 13.4894 13.7006 13.4894H3.20268C2.75277 13.4894 2.45283 13.1895 2.45283 12.7396V8.24048C2.45283 7.79057 2.15289 7.49063 1.70298 7.49063C1.25307 7.49063 0.953125 7.79057 0.953125 8.24048V12.7396C0.953125 14.0143 1.92793 14.9891 3.20268 14.9891H13.7006C14.9753 14.9891 15.9501 14.0143 15.9501 12.7396V8.24048C15.9501 7.79057 15.6502 7.49063 15.2003 7.49063ZM5.97713 4.26627L7.70178 2.54161V9.74018C7.70178 10.1901 8.00172 10.49 8.45163 10.49C8.90155 10.49 9.20149 10.1901 9.20149 9.74018V2.54161L10.9261 4.26627C11.2261 4.56621 11.676 4.56621 11.9759 4.26627C12.2759 3.96633 12.2759 3.51642 11.9759 3.21648L8.97653 0.217073C8.90155 0.142088 8.82656 0.0671031 8.75157 0.0671031C8.6016 -0.00788202 8.37665 -0.00788202 8.15169 0.0671031C8.07671 0.0671031 8.00172 0.142088 7.92674 0.217073L4.92734 3.21648C4.62739 3.51642 4.62739 3.96633 4.92734 4.26627C5.22728 4.56621 5.67719 4.56621 5.97713 4.26627Z",
      fill: "#A0A3AD"
    })
  });
}
const Ee = ({
  url: e
}) => {
  const [t, s] = V.useState(!1), o = async (i) => {
    try {
      await navigator.clipboard.writeText(e), s(!0), setTimeout(() => {
        s(!1);
      }, 2e3);
    } catch (r) {
      console.error("Failed to copy the text to clipboard:", r);
    }
  };
  return /* @__PURE__ */ n.jsx("div", {
    children: t ? /* @__PURE__ */ n.jsx("svg", {
      style: {
        cursor: "pointer",
        fill: "green"
      },
      width: "16",
      height: "15",
      viewBox: "0 0 16 15",
      xmlns: "http://www.w3.org/2000/svg",
      children: /* @__PURE__ */ n.jsx("path", {
        d: "M5.99967 12.8136L1.71967 8.53364L0.559673 9.69364L5.99967 15.1336L16.4397 4.69364L15.2797 3.53364L5.99967 12.8136Z"
      })
    }) : /* @__PURE__ */ n.jsx(un, {
      onClick: o
    })
  });
}, dn = () => /* @__PURE__ */ n.jsxs("svg", {
  width: "10",
  height: "10",
  viewBox: "0 0 10 10",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  children: [/* @__PURE__ */ n.jsx("path", {
    d: "M5.85855 0.555664H8.31281M8.31281 0.555664V2.73754M8.31281 0.555664L4.31445 4.11122",
    stroke: "#FFFFFF",
    "stroke-linecap": "round",
    "stroke-linejoin": "round"
  }), /* @__PURE__ */ n.jsx("path", {
    "fill-rule": "evenodd",
    "clip-rule": "evenodd",
    d: "M3.31445 1H1.31445C0.762168 1 0.314453 1.44772 0.314453 2V8C0.314453 8.55228 0.762168 9 1.31445 9H7.31445C7.86674 9 8.31445 8.55228 8.31445 8V6H7.31445V8H1.31445V2H3.31445V1Z",
    fill: "#FFFFFF"
  })]
}), fn = () => /* @__PURE__ */ n.jsx("svg", {
  width: "10",
  height: "14",
  viewBox: "0 0 10 14",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  children: /* @__PURE__ */ n.jsx("path", {
    d: "M5.01451 12.5187L4.96833 12.5174L4.93914 12.522C4.93113 12.516 4.92402 12.5087 4.91808 12.5005L4.51231 12.7926L4.91808 12.5005C4.9049 12.4822 4.89824 12.46 4.89913 12.4375L4.89953 12.4275V12.4176V8.81194V8.31194H4.39953H1.8876H1.88753C1.76188 8.31196 1.63853 8.27825 1.53036 8.21433C1.42218 8.1504 1.33315 8.05862 1.27256 7.94854C1.21196 7.83847 1.18203 7.71415 1.18588 7.58856C1.18973 7.46301 1.2272 7.34079 1.29438 7.23466C1.2944 7.23463 1.29442 7.23459 1.29444 7.23456L4.90001 1.54377L4.90567 1.53483L4.91095 1.52567C4.92227 1.50601 4.93988 1.49074 4.96094 1.48232C4.982 1.47389 5.00528 1.47281 5.02703 1.47924L5.02912 1.47985C5.05077 1.48614 5.06969 1.4995 5.08286 1.5178C5.09603 1.53609 5.1027 1.55827 5.1018 1.58079L5.10141 1.59073V1.60067V5.20631V5.70631H5.60141H8.11333H8.1134C8.23905 5.70629 8.3624 5.74 8.47058 5.80392C8.57875 5.86784 8.66778 5.95963 8.72838 6.06971C8.78897 6.17978 8.81891 6.3041 8.81506 6.42969C8.81121 6.55528 8.77371 6.67753 8.70649 6.78369L5.10232 12.4723C5.10219 12.4725 5.10205 12.4727 5.10191 12.4729C5.09255 12.4873 5.07969 12.499 5.06452 12.507C5.04914 12.5152 5.03191 12.5192 5.01451 12.5187Z",
    stroke: "#FFFFFF"
  })
}), hn = /* @__PURE__ */ g("button")({
  classes: ["f1hmcrrf"]
}), Cn = /* @__PURE__ */ g("div")({
  classes: ["f1ki4il6"]
}), pn = /* @__PURE__ */ g("div")({
  classes: ["bgnz0ud"]
}), gn = /* @__PURE__ */ g("img")({
  classes: ["f1gbget9"]
}), mn = /* @__PURE__ */ g("div")({
  classes: ["f4rwe4i"]
}), xn = /* @__PURE__ */ g("div")({
  classes: ["f1qellgw"]
}), wn = /* @__PURE__ */ g("input")({
  classes: ["i1hgxwhi"]
}), vn = /* @__PURE__ */ g("div")({
  classes: ["s1ahcfax"]
}), jn = () => {
  const e = Y(null);
  return H(() => {
    if (e.current) {
      let t = 0;
      const s = () => {
        t += 6, e.current && (e.current.style.transform = `rotate(${t}deg)`), requestAnimationFrame(s);
      };
      requestAnimationFrame(s);
    }
  }, []), /* @__PURE__ */ n.jsx("svg", {
    ref: e,
    xmlns: "http://www.w3.org/2000/svg",
    fill: "none",
    viewBox: "0 0 24 24",
    strokeWidth: "1.5",
    stroke: "currentColor",
    className: "size-6 text-white",
    style: {
      width: "24px",
      height: "24px"
    },
    children: /* @__PURE__ */ n.jsx("path", {
      strokeLinecap: "round",
      strokeLinejoin: "round",
      d: "M16.023 9.348h4.992v-.001M2.985 19.644v-4.992m0 0h4.992m-4.993 0 3.181 3.183a8.25 8.25 0 0 0 13.803-3.7M4.031 9.865a8.25 8.25 0 0 1 13.803-3.7l3.181 3.182m0-4.991v4.99"
    })
  });
};
function yn({
  number: e,
  text: t,
  actionType: s,
  target: o,
  frameUrl: i,
  handleOnClick: r
}) {
  return /* @__PURE__ */ n.jsxs(hn, {
    onClick: () => r(e),
    children: [t, (s === "link" || s === "post_redirect" || s === "mint") && /* @__PURE__ */ n.jsx(dn, {}), s === "tx" && /* @__PURE__ */ n.jsx(fn, {})]
  });
}
function En({
  frame: e,
  onFrameBtnPress: t
}) {
  var C;
  const [s, o] = N(e), [i, r] = N(""), [a, u] = N(!1), d = () => {
    const c = s.buttons.map((x) => /* @__PURE__ */ n.jsx(yn, {
      number: x.index,
      text: x.title,
      actionType: x.action_type,
      target: x.target,
      frameUrl: e.frames_url,
      handleOnClick: (p) => {
        u(!0), t(p, s, o, i).finally(() => u(!1));
      }
    }, x.index));
    return /* @__PURE__ */ n.jsx(pn, {
      children: c
    });
  }, l = (c) => {
    r(c);
  }, m = (c) => {
    try {
      return new URL(c).hostname.replace("www.", "");
    } catch {
      return "";
    }
  }, f = () => {
    switch (s.image_aspect_ratio) {
      case "1:1":
        return {
          aspectRatio: "1 / 1"
        };
      case "1.91:1":
        return {
          aspectRatio: "1.91 / 1"
        };
      default:
        return {
          aspectRatio: "1.91 / 1"
        };
    }
  };
  return /* @__PURE__ */ n.jsxs(n.Fragment, {
    children: [/* @__PURE__ */ n.jsxs(Cn, {
      children: [a && /* @__PURE__ */ n.jsx(vn, {
        children: /* @__PURE__ */ n.jsx(jn, {})
      }), s.frames_url && /* @__PURE__ */ n.jsxs(n.Fragment, {
        children: [/* @__PURE__ */ n.jsx("a", {
          href: s.frames_url,
          target: "_blank",
          rel: "noopener noreferrer",
          style: {
            width: "100%"
          },
          children: /* @__PURE__ */ n.jsx(gn, {
            src: s.image,
            alt: `Frame image for ${s.frames_url}`,
            style: f()
          })
        }), ((C = s.input) == null ? void 0 : C.text) && /* @__PURE__ */ n.jsx(wn, {
          type: "text",
          placeholder: s.input.text,
          value: i,
          onChange: (c) => l(c.target.value)
        }), d()]
      })]
    }), s.frames_url && /* @__PURE__ */ n.jsx(mn, {
      children: m(s.frames_url)
    })]
  });
}
const _n = ({
  frame: e,
  onFrameBtnPress: t
}) => /* @__PURE__ */ n.jsx(xn, {
  children: e ? /* @__PURE__ */ n.jsx(En, {
    frame: e,
    onFrameBtnPress: t
  }) : /* @__PURE__ */ n.jsx(n.Fragment, {})
}), Rn = ({
  url: e,
  onFrameBtnPress: t,
  initialFrame: s
}) => {
  const {
    client_id: o,
    showToast: i
  } = o1(), [r] = le(w1.NEYNAR_AUTHENTICATED_USER, null), [a, u] = N(null), [d, l] = N(s || null), [m, f] = N(null);
  H(() => {
    r ? u(r.signer_uuid) : console.warn("No NEYNAR_AUTHENTICATED_USER found in local storage.");
  }, [r]), H(() => {
    s || (async () => {
      try {
        const p = await Sn(`${C1}/v2/farcaster/frame/crawl?url=${e}&client_id=${o}`, {
          method: "GET"
        });
        if (p.ok) {
          const v = (await p.json()).frame;
          if (Object.keys(v).length === 0)
            throw new Error("No frame data available");
          l(v), f(null);
        } else
          throw new Error(`HTTP error! status: ${p.status}`);
      } catch (p) {
        console.error(`An error occurred: ${p}`), f(`Failed to fetch: ${p.message}`);
      }
    })();
  }, [e, i, s]);
  const C = (x) => {
    if (typeof x != "object" || x === null) return !1;
    const p = ["version", "image", "buttons", "frames_url"];
    for (const y of p)
      if (!(y in x)) return !1;
    return !(!Array.isArray(x.buttons) || x.buttons.some((y) => typeof y.index != "number"));
  }, c = async (x, p, y, v) => {
    try {
      const S = await t(x, p, y, v);
      if (!C(S))
        throw new Error("Invalid frame data received");
      y(S);
    } catch (S) {
      i(Be.Error, `An error occurred while processing the button press: ${S}`);
    }
  };
  return m ? /* @__PURE__ */ n.jsx("div", {
    style: {
      color: "red",
      padding: "10px",
      border: "1px solid red",
      borderRadius: "5px"
    },
    children: m
  }) : /* @__PURE__ */ n.jsx(_n, {
    frame: d,
    onFrameBtnPress: c
  });
};
function Sn(e, t, s = 8e3) {
  return Promise.race([p1(e, t), new Promise((o, i) => setTimeout(() => i(new Error("Request timed out")), s))]);
}
const An = /* @__PURE__ */ g("div")({
  classes: ["sxqvxvq"]
}), bn = /* @__PURE__ */ g("a")({
  classes: ["s3kamov"]
}), Ln = /* @__PURE__ */ g("div")({
  classes: ["m1wrvvhk"]
}), In = /* @__PURE__ */ g("div")({
  classes: ["u1lcaq27"]
}), Tn = /* @__PURE__ */ g("div")({
  classes: ["lvnxx7k"]
}), _e = /* @__PURE__ */ g("div")({
  classes: ["e1dqpn4e"]
}), Vn = /* @__PURE__ */ g("div")({
  classes: ["r15yf123"]
}), Re = /* @__PURE__ */ g("div")({
  classes: ["skmprsx"]
}), $1 = V.memo(({
  username: e,
  displayName: t,
  avatarImgUrl: s,
  text: o = "",
  hash: i,
  reactions: r,
  replies: a,
  embeds: u = [],
  frames: d = [],
  channel: l,
  viewerFid: m,
  hasPowerBadge: f,
  isEmbed: C = !0,
  allowReactions: c,
  renderEmbeds: x,
  renderFrames: p,
  onLikeBtnPress: y,
  onRecastBtnPress: v,
  onCommentBtnPress: S,
  onFrameBtnPress: E,
  direct_replies: R,
  containerStyles: h,
  textStyles: w
}) => {
  const [I, j] = N(r.likes_count), [T, A] = N(r.likes.some((b) => b.fid === m)), F = Dt(o, u), z = (u == null ? void 0 : u.length) === 1, D = f1(() => d.map((b) => b.frames_url), [d]), L = f1(() => u.filter((b) => !D.includes(b.url)), [u, D]), P = J((b) => {
    b.currentTarget.src = E1;
  }, []);
  H(() => {
    A(r.likes.some((b) => b.fid === m));
  }, [r.likes, m]);
  const q = J(() => y && y() ? (j((t1) => t1 + 1), A(!T), !0) : !1, [y]), O = Jt(L, c, m);
  return /* @__PURE__ */ n.jsx(An, {
    style: {
      ...h,
      borderWidth: C ? "1px" : "0"
    },
    children: /* @__PURE__ */ n.jsxs(X, {
      children: [/* @__PURE__ */ n.jsx(K, {
        spacingRight: "10px",
        children: /* @__PURE__ */ n.jsx(N1, {
          src: s && s.length > 0 ? s : E1,
          onError: P,
          loading: "lazy",
          alt: `${t ?? "Skeleton"} Avatar`
        })
      }), /* @__PURE__ */ n.jsxs(Ln, {
        children: [/* @__PURE__ */ n.jsx(X, {
          justifyContent: "space-between",
          flexGrow: 1,
          children: /* @__PURE__ */ n.jsxs(Je, {
            children: [/* @__PURE__ */ n.jsxs(X, {
              children: [/* @__PURE__ */ n.jsx("strong", {
                children: t
              }), f && /* @__PURE__ */ n.jsx(K, {
                spacingLeft: "5px",
                children: /* @__PURE__ */ n.jsx(Ke, {})
              })]
            }), /* @__PURE__ */ n.jsx(X, {
              alignItems: "center",
              children: /* @__PURE__ */ n.jsxs(In, {
                children: ["@", e]
              })
            })]
          })
        }), /* @__PURE__ */ n.jsx(K, {
          spacingVertical: "15px",
          children: /* @__PURE__ */ n.jsx(Tn, {
            style: w,
            children: F
          })
        }), x && L && L.length > 0 ? /* @__PURE__ */ n.jsx(_e, {
          style: {
            margin: z ? "10px 0" : "0"
          },
          children: O.map((b, t1) => /* @__PURE__ */ n.jsx("div", {
            style: {
              width: "100%"
            },
            children: b
          }, t1))
        }) : /* @__PURE__ */ n.jsx(n.Fragment, {}), p && d && d.length > 0 ? /* @__PURE__ */ n.jsx(_e, {
          children: d.map((b) => /* @__PURE__ */ n.jsx(Rn, {
            url: b.frames_url,
            initialFrame: b,
            onFrameBtnPress: E
          }, b.frames_url))
        }) : null, /* @__PURE__ */ n.jsxs(Vn, {
          style: {
            justifyContent: c ? "space-between" : "flex-end"
          },
          children: [c && /* @__PURE__ */ n.jsx(ln, {
            hash: i,
            reactions: r,
            onComment: S,
            onRecast: v,
            onLike: q,
            isLiked: T
          }), c && e && i && /* @__PURE__ */ n.jsx(Ee, {
            url: `https://warpcast.com/${e}/${i.slice(0, 10)}`
          })]
        }), /* @__PURE__ */ n.jsxs(Re, {
          style: {
            justifyContent: c ? "" : "space-between"
          },
          children: [/* @__PURE__ */ n.jsxs(Re, {
            style: {
              justifyContent: c ? "" : "space-between",
              gap: 6
            },
            children: [/* @__PURE__ */ n.jsxs("div", {
              children: [a, " replies"]
            }), /* @__PURE__ */ n.jsx("div", {
              children: ""
            }), /* @__PURE__ */ n.jsxs("div", {
              children: [I, " likes"]
            }), l && /* @__PURE__ */ n.jsxs(n.Fragment, {
              children: [/* @__PURE__ */ n.jsx("div", {
                children: ""
              }), /* @__PURE__ */ n.jsxs(bn, {
                href: `https://warpcast.com/~/channel/${l.id}`,
                target: "_blank",
                children: ["/", l.id]
              })]
            })]
          }), !c && e && i && /* @__PURE__ */ n.jsx(Ee, {
            url: `https://warpcast.com/${e}/${i.slice(0, 10)}`
          })]
        })]
      })]
    })
  });
}), Nn = /* @__PURE__ */ g("div")({
  classes: ["s1m6ob1a"]
}), Se = /* @__PURE__ */ g("div")({
  classes: ["r1esevc7"]
}), Ae = /* @__PURE__ */ g("div")({
  classes: ["vxb1f00"]
}), $n = /* @__PURE__ */ g("div")({
  classes: ["htv0qz7"]
}), be = /* @__PURE__ */ g("div")({
  classes: ["rzksy2"]
});
function kn(e) {
  return /* @__PURE__ */ n.jsx(Nn, {
    children: e.casts.map((t, s) => /* @__PURE__ */ n.jsxs(Se, {
      children: [s !== 0 && /* @__PURE__ */ n.jsx(Ae, {
        style: {
          left: "42px"
        }
      }), /* @__PURE__ */ n.jsxs(be, {
        children: [/* @__PURE__ */ n.jsx($1, {
          isEmbed: !1,
          ...t
        }), s === 0 && /* @__PURE__ */ n.jsx($n, {}), t.direct_replies && t.direct_replies.length > 0 && t.direct_replies.map((o, i) => /* @__PURE__ */ n.jsxs(Se, {
          children: [/* @__PURE__ */ n.jsx(Ae, {
            style: {
              left: "42px"
            }
          }), /* @__PURE__ */ n.jsx(be, {
            children: /* @__PURE__ */ n.jsx($1, {
              isEmbed: !1,
              ...o
            })
          })]
        }, i))]
      })]
    }, s))
  });
}
async function Dn({
  type: e,
  identifier: t,
  replyDepth: s = 2,
  includeChronologicalParentCasts: o = !1,
  limit: i = 20,
  viewerFid: r,
  clientId: a
}) {
  try {
    let u = `${C1}/v2/farcaster/cast/conversation?identifier=${encodeURIComponent(t)}&type=${e}&reply_depth=${s}&include_chronological_parent_casts=${o}&limit=${i}&client_id=${a}`;
    return r && (u += `&viewer_fid=${r}`), await (await p1(u, {
      method: "GET",
      headers: {
        accept: "application/json"
      }
    })).json() || null;
  } catch (u) {
    return console.error("Error fetching conversation", u), null;
  }
}
function tt(e) {
  return {
    username: e.author.username,
    displayName: e.author.display_name,
    avatarImgUrl: e.author.pfp_url,
    text: e.text,
    hash: e.hash,
    reactions: e.reactions,
    replies: e.replies.count,
    embeds: e.embeds,
    frames: e.frames,
    renderEmbeds: e.renderEmbeds,
    channel: e.channel,
    viewerFid: 2,
    hasPowerBadge: e.author.power_badge,
    isOwnProfile: !1,
    allowReactions: !0,
    renderFrames: !1,
    direct_replies: e.direct_replies ? e.direct_replies.map(tt) : []
  };
}
function Fn(e) {
  const t = [];
  return e.cast && t.push(tt(e.cast)), t;
}
const gr = ({
  type: e,
  identifier: t,
  replyDepth: s = 2,
  includeChronologicalParentCasts: o = !1,
  limit: i = 20,
  viewerFid: r
}) => {
  const {
    client_id: a
  } = o1(), [u, d] = V.useState(null), [l, m] = V.useState(!0), [f, C] = V.useState(null);
  return V.useEffect(() => {
    m(!0), C(null), Dn({
      type: e,
      identifier: t,
      replyDepth: s,
      includeChronologicalParentCasts: o,
      limit: i,
      viewerFid: r,
      clientId: a
    }).then((c) => {
      d(c);
    }).catch((c) => {
      C(c);
    }).finally(() => {
      m(!1);
    });
  }, [e, t, s, o, i, r, a]), l ? /* @__PURE__ */ n.jsx(n.Fragment, {
    children: " "
  }) : f ? /* @__PURE__ */ n.jsx(n.Fragment, {
    children: " "
  }) : /* @__PURE__ */ n.jsx(kn, {
    casts: Fn(u.conversation)
  });
};
var nt = { exports: {} }, U1 = {};
/**
 * @license React
 * use-sync-external-store-shim.production.js
 *
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Le;
function Mn() {
  if (Le) return U1;
  Le = 1;
  var e = V;
  function t(f, C) {
    return f === C && (f !== 0 || 1 / f === 1 / C) || f !== f && C !== C;
  }
  var s = typeof Object.is == "function" ? Object.is : t, o = e.useState, i = e.useEffect, r = e.useLayoutEffect, a = e.useDebugValue;
  function u(f, C) {
    var c = C(), x = o({ inst: { value: c, getSnapshot: C } }), p = x[0].inst, y = x[1];
    return r(
      function() {
        p.value = c, p.getSnapshot = C, d(p) && y({ inst: p });
      },
      [f, c, C]
    ), i(
      function() {
        return d(p) && y({ inst: p }), f(function() {
          d(p) && y({ inst: p });
        });
      },
      [f]
    ), a(c), c;
  }
  function d(f) {
    var C = f.getSnapshot;
    f = f.value;
    try {
      var c = C();
      return !s(f, c);
    } catch {
      return !0;
    }
  }
  function l(f, C) {
    return C();
  }
  var m = typeof window > "u" || typeof window.document > "u" || typeof window.document.createElement > "u" ? l : u;
  return U1.useSyncExternalStore = e.useSyncExternalStore !== void 0 ? e.useSyncExternalStore : m, U1;
}
nt.exports = Mn();
var st = nt.exports;
const rt = 0, ot = 1, it = 2, Ie = 3;
var Te = Object.prototype.hasOwnProperty;
function K1(e, t) {
  var s, o;
  if (e === t) return !0;
  if (e && t && (s = e.constructor) === t.constructor) {
    if (s === Date) return e.getTime() === t.getTime();
    if (s === RegExp) return e.toString() === t.toString();
    if (s === Array) {
      if ((o = e.length) === t.length)
        for (; o-- && K1(e[o], t[o]); ) ;
      return o === -1;
    }
    if (!s || typeof e == "object") {
      o = 0;
      for (s in e)
        if (Te.call(e, s) && ++o && !Te.call(t, s) || !(s in t) || !K1(e[s], t[s])) return !1;
      return Object.keys(t).length === o;
    }
  }
  return e !== e && t !== t;
}
const r1 = /* @__PURE__ */ new WeakMap(), d1 = () => {
}, U = (
  /*#__NOINLINE__*/
  d1()
), Q1 = Object, _ = (e) => e === U, e1 = (e) => typeof e == "function", h1 = (e, t) => ({
  ...e,
  ...t
}), at = (e) => e1(e.then), P1 = {}, L1 = {}, ue = "undefined", S1 = typeof window != ue, X1 = typeof document != ue, On = S1 && "Deno" in window, Hn = () => S1 && typeof window.requestAnimationFrame != ue, u1 = (e, t) => {
  const s = r1.get(e);
  return [
    // Getter
    () => !_(t) && e.get(t) || P1,
    // Setter
    (o) => {
      if (!_(t)) {
        const i = e.get(t);
        t in L1 || (L1[t] = i), s[5](t, h1(i, o), i || P1);
      }
    },
    // Subscriber
    s[6],
    // Get server cache snapshot
    () => !_(t) && t in L1 ? L1[t] : !_(t) && e.get(t) || P1
  ];
};
let ee = !0;
const Un = () => ee, [te, ne] = S1 && window.addEventListener ? [
  window.addEventListener.bind(window),
  window.removeEventListener.bind(window)
] : [
  d1,
  d1
], Pn = () => {
  const e = X1 && document.visibilityState;
  return _(e) || e !== "hidden";
}, qn = (e) => (X1 && document.addEventListener("visibilitychange", e), te("focus", e), () => {
  X1 && document.removeEventListener("visibilitychange", e), ne("focus", e);
}), Wn = (e) => {
  const t = () => {
    ee = !0, e();
  }, s = () => {
    ee = !1;
  };
  return te("online", t), te("offline", s), () => {
    ne("online", t), ne("offline", s);
  };
}, Bn = {
  isOnline: Un,
  isVisible: Pn
}, zn = {
  initFocus: qn,
  initReconnect: Wn
}, Ve = !V.useId, _1 = !S1 || On, Zn = (e) => Hn() ? window.requestAnimationFrame(e) : setTimeout(e, 1), V1 = _1 ? H : xt, q1 = typeof navigator < "u" && navigator.connection, Ne = !_1 && q1 && ([
  "slow-2g",
  "2g"
].includes(q1.effectiveType) || q1.saveData), I1 = /* @__PURE__ */ new WeakMap(), W1 = (e, t) => Q1.prototype.toString.call(e) === `[object ${t}]`;
let Yn = 0;
const se = (e) => {
  const t = typeof e, s = W1(e, "Date"), o = W1(e, "RegExp"), i = W1(e, "Object");
  let r, a;
  if (Q1(e) === e && !s && !o) {
    if (r = I1.get(e), r) return r;
    if (r = ++Yn + "~", I1.set(e, r), Array.isArray(e)) {
      for (r = "@", a = 0; a < e.length; a++)
        r += se(e[a]) + ",";
      I1.set(e, r);
    }
    if (i) {
      r = "#";
      const u = Q1.keys(e).sort();
      for (; !_(a = u.pop()); )
        _(e[a]) || (r += a + ":" + se(e[a]) + ",");
      I1.set(e, r);
    }
  } else
    r = s ? e.toJSON() : t == "symbol" ? e.toString() : t == "string" ? JSON.stringify(e) : "" + e;
  return r;
}, R1 = (e) => {
  if (e1(e))
    try {
      e = e();
    } catch {
      e = "";
    }
  const t = e;
  return e = typeof e == "string" ? e : (Array.isArray(e) ? e.length : e) ? se(e) : "", [
    e,
    t
  ];
};
let Gn = 0;
const re = () => ++Gn;
async function ct(...e) {
  const [t, s, o, i] = e, r = h1({
    populateCache: !0,
    throwOnError: !0
  }, typeof i == "boolean" ? {
    revalidate: i
  } : i || {});
  let a = r.populateCache;
  const u = r.rollbackOnError;
  let d = r.optimisticData;
  const l = (C) => typeof u == "function" ? u(C) : u !== !1, m = r.throwOnError;
  if (e1(s)) {
    const C = s, c = [], x = t.keys();
    for (const p of x)
      // Skip the special useSWRInfinite and useSWRSubscription keys.
      !/^\$(inf|sub)\$/.test(p) && C(t.get(p)._k) && c.push(p);
    return Promise.all(c.map(f));
  }
  return f(s);
  async function f(C) {
    const [c] = R1(C);
    if (!c) return;
    const [x, p] = u1(t, c), [y, v, S, E] = r1.get(t), R = () => {
      const D = y[c];
      return (e1(r.revalidate) ? r.revalidate(x().data, C) : r.revalidate !== !1) && (delete S[c], delete E[c], D && D[0]) ? D[0](it).then(() => x().data) : x().data;
    };
    if (e.length < 3)
      return R();
    let h = o, w;
    const I = re();
    v[c] = [
      I,
      0
    ];
    const j = !_(d), T = x(), A = T.data, F = T._c, z = _(F) ? A : F;
    if (j && (d = e1(d) ? d(z, A) : d, p({
      data: d,
      _c: z
    })), e1(h))
      try {
        h = h(z);
      } catch (D) {
        w = D;
      }
    if (h && at(h))
      if (h = await h.catch((D) => {
        w = D;
      }), I !== v[c][0]) {
        if (w) throw w;
        return h;
      } else w && j && l(w) && (a = !0, p({
        data: z,
        _c: U
      }));
    if (a && !w)
      if (e1(a)) {
        const D = a(h, z);
        p({
          data: D,
          error: U,
          _c: U
        });
      } else
        p({
          data: h,
          error: U,
          _c: U
        });
    if (v[c][1] = re(), Promise.resolve(R()).then(() => {
      p({
        _c: U
      });
    }), w) {
      if (m) throw w;
      return;
    }
    return h;
  }
}
const $e = (e, t) => {
  for (const s in e)
    e[s][0] && e[s][0](t);
}, Jn = (e, t) => {
  if (!r1.has(e)) {
    const s = h1(zn, t), o = {}, i = ct.bind(U, e);
    let r = d1;
    const a = {}, u = (m, f) => {
      const C = a[m] || [];
      return a[m] = C, C.push(f), () => C.splice(C.indexOf(f), 1);
    }, d = (m, f, C) => {
      e.set(m, f);
      const c = a[m];
      if (c)
        for (const x of c)
          x(f, C);
    }, l = () => {
      if (!r1.has(e) && (r1.set(e, [
        o,
        {},
        {},
        {},
        i,
        d,
        u
      ]), !_1)) {
        const m = s.initFocus(setTimeout.bind(U, $e.bind(U, o, rt))), f = s.initReconnect(setTimeout.bind(U, $e.bind(U, o, ot)));
        r = () => {
          m && m(), f && f(), r1.delete(e);
        };
      }
    };
    return l(), [
      e,
      i,
      l,
      r
    ];
  }
  return [
    e,
    r1.get(e)[4]
  ];
}, Kn = (e, t, s, o, i) => {
  const r = s.errorRetryCount, a = i.retryCount, u = ~~((Math.random() + 0.5) * (1 << (a < 8 ? a : 8))) * s.errorRetryInterval;
  !_(r) && a > r || setTimeout(o, u, i);
}, Qn = K1, [de, Xn] = Jn(/* @__PURE__ */ new Map()), es = h1(
  {
    // events
    onLoadingSlow: d1,
    onSuccess: d1,
    onError: d1,
    onErrorRetry: Kn,
    onDiscarded: d1,
    // switches
    revalidateOnFocus: !0,
    revalidateOnReconnect: !0,
    revalidateIfStale: !0,
    shouldRetryOnError: !0,
    // timeouts
    errorRetryInterval: Ne ? 1e4 : 5e3,
    focusThrottleInterval: 5 * 1e3,
    dedupingInterval: 2 * 1e3,
    loadingTimeout: Ne ? 5e3 : 3e3,
    // providers
    compare: Qn,
    isPaused: () => !1,
    cache: de,
    mutate: Xn,
    fallback: {}
  },
  // use web preset by default
  Bn
), ts = (e, t) => {
  const s = h1(e, t);
  if (t) {
    const { use: o, fallback: i } = e, { use: r, fallback: a } = t;
    o && r && (s.use = o.concat(r)), i && a && (s.fallback = h1(i, a));
  }
  return s;
}, ns = ae({}), lt = "$inf$", ut = S1 && window.__SWR_DEVTOOLS_USE__, ss = ut ? window.__SWR_DEVTOOLS_USE__ : [], rs = () => {
  ut && (window.__SWR_DEVTOOLS_REACT__ = V);
}, dt = (e) => e1(e[1]) ? [
  e[0],
  e[1],
  e[2] || {}
] : [
  e[0],
  null,
  (e[1] === null ? e[2] : e[1]) || {}
], os = () => h1(es, ce(ns)), is = (e) => (t, s, o) => e(t, s && ((...r) => {
  const [a] = R1(t), [, , , u] = r1.get(de);
  if (a.startsWith(lt))
    return s(...r);
  const d = u[a];
  return _(d) ? s(...r) : (delete u[a], d);
}), o), as = ss.concat(is), cs = (e) => function(...s) {
  const o = os(), [i, r, a] = dt(s), u = ts(o, a);
  let d = e;
  const { use: l } = u, m = (l || []).concat(as);
  for (let f = m.length; f--; )
    d = m[f](d);
  return d(i, r || u.fetcher || null, u);
}, ls = (e, t, s) => {
  const o = t[e] || (t[e] = []);
  return o.push(s), () => {
    const i = o.indexOf(s);
    i >= 0 && (o[i] = o[o.length - 1], o.pop());
  };
}, us = (e, t) => (...s) => {
  const [o, i, r] = dt(s), a = (r.use || []).concat(t);
  return e(o, i, {
    ...r,
    use: a
  });
};
rs();
const B1 = V.use || // This extra generic is to avoid TypeScript mixing up the generic and JSX sytax
// and emitting an error.
// We assume that this is only for the `use(thenable)` case, not `use(context)`.
// https://github.com/facebook/react/blob/aed00dacfb79d17c53218404c52b1c7aa59c4a89/packages/react-server/src/ReactFizzThenable.js#L45
((e) => {
  switch (e.status) {
    case "pending":
      throw e;
    case "fulfilled":
      return e.value;
    case "rejected":
      throw e.reason;
    default:
      throw e.status = "pending", e.then((t) => {
        e.status = "fulfilled", e.value = t;
      }, (t) => {
        e.status = "rejected", e.reason = t;
      }), e;
  }
}), z1 = {
  dedupe: !0
}, ds = (e, t, s) => {
  const { cache: o, compare: i, suspense: r, fallbackData: a, revalidateOnMount: u, revalidateIfStale: d, refreshInterval: l, refreshWhenHidden: m, refreshWhenOffline: f, keepPreviousData: C } = s, [c, x, p, y] = r1.get(o), [v, S] = R1(e), E = Y(!1), R = Y(!1), h = Y(v), w = Y(t), I = Y(s), j = () => I.current, T = () => j().isVisible() && j().isOnline(), [A, F, z, D] = u1(o, v), L = Y({}).current, P = _(a) ? _(s.fallback) ? U : s.fallback[v] : a, q = ($, k) => {
    for (const Z in L) {
      const M = Z;
      if (M === "data") {
        if (!i($[M], k[M]) && (!_($[M]) || !i(v1, k[M])))
          return !1;
      } else if (k[M] !== $[M])
        return !1;
    }
    return !0;
  }, O = f1(() => {
    const $ = !v || !t ? !1 : _(u) ? j().isPaused() || r ? !1 : d !== !1 : u, k = (G) => {
      const i1 = h1(G);
      return delete i1._k, $ ? {
        isValidating: !0,
        isLoading: !0,
        ...i1
      } : i1;
    }, Z = A(), M = D(), n1 = k(Z), x1 = Z === M ? n1 : k(M);
    let B = n1;
    return [
      () => {
        const G = k(A());
        return q(G, B) ? (B.data = G.data, B.isLoading = G.isLoading, B.isValidating = G.isValidating, B.error = G.error, B) : (B = G, G);
      },
      () => x1
    ];
  }, [
    o,
    v
  ]), b = st.useSyncExternalStore(J(
    ($) => z(v, (k, Z) => {
      q(Z, k) || $();
    }),
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [
      o,
      v
    ]
  ), O[0], O[1]), t1 = !E.current, k1 = c[v] && c[v].length > 0, l1 = b.data, W = _(l1) ? P && at(P) ? B1(P) : P : l1, g1 = b.error, m1 = Y(W), v1 = C ? _(l1) ? _(m1.current) ? W : m1.current : l1 : W, A1 = k1 && !_(g1) ? !1 : t1 && !_(u) ? u : j().isPaused() ? !1 : r ? _(W) ? !1 : d : _(W) || d, fe = !!(v && t && t1 && A1), pt = _(b.isValidating) ? fe : b.isValidating, gt = _(b.isLoading) ? fe : b.isLoading, j1 = J(
    async ($) => {
      const k = w.current;
      if (!v || !k || R.current || j().isPaused())
        return !1;
      let Z, M, n1 = !0;
      const x1 = $ || {}, B = !p[v] || !x1.dedupe, G = () => Ve ? !R.current && v === h.current && E.current : v === h.current, i1 = {
        isValidating: !1,
        isLoading: !1
      }, Ce = () => {
        F(i1);
      }, pe = () => {
        const Q = p[v];
        Q && Q[1] === M && delete p[v];
      }, ge = {
        isValidating: !0
      };
      _(A().data) && (ge.isLoading = !0);
      try {
        if (B && (F(ge), s.loadingTimeout && _(A().data) && setTimeout(() => {
          n1 && G() && j().onLoadingSlow(v, s);
        }, s.loadingTimeout), p[v] = [
          k(S),
          re()
        ]), [Z, M] = p[v], Z = await Z, B && setTimeout(pe, s.dedupingInterval), !p[v] || p[v][1] !== M)
          return B && G() && j().onDiscarded(v), !1;
        i1.error = U;
        const Q = x[v];
        if (!_(Q) && // case 1
        (M <= Q[0] || // case 2
        M <= Q[1] || // case 3
        Q[1] === 0))
          return Ce(), B && G() && j().onDiscarded(v), !1;
        const a1 = A().data;
        i1.data = i(a1, Z) ? a1 : Z, B && G() && j().onSuccess(Z, v, s);
      } catch (Q) {
        pe();
        const a1 = j(), { shouldRetryOnError: D1 } = a1;
        a1.isPaused() || (i1.error = Q, B && G() && (a1.onError(Q, v, a1), (D1 === !0 || e1(D1) && D1(Q)) && (!j().revalidateOnFocus || !j().revalidateOnReconnect || T()) && a1.onErrorRetry(Q, v, a1, (mt) => {
          const F1 = c[v];
          F1 && F1[0] && F1[0](Ie, mt);
        }, {
          retryCount: (x1.retryCount || 0) + 1,
          dedupe: !0
        })));
      }
      return n1 = !1, Ce(), !0;
    },
    // `setState` is immutable, and `eventsCallback`, `fnArg`, and
    // `keyValidating` are depending on `key`, so we can exclude them from
    // the deps array.
    //
    // FIXME:
    // `fn` and `config` might be changed during the lifecycle,
    // but they might be changed every render like this.
    // `useSWR('key', () => fetch('/api/'), { suspense: true })`
    // So we omit the values from the deps array
    // even though it might cause unexpected behaviors.
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [
      v,
      o
    ]
  ), he = J(
    // Use callback to make sure `keyRef.current` returns latest result every time
    (...$) => ct(o, h.current, ...$),
    // eslint-disable-next-line react-hooks/exhaustive-deps
    []
  );
  if (V1(() => {
    w.current = t, I.current = s, _(l1) || (m1.current = l1);
  }), V1(() => {
    if (!v) return;
    const $ = j1.bind(U, z1);
    let k = 0;
    const M = ls(v, c, (n1, x1 = {}) => {
      if (n1 == rt) {
        const B = Date.now();
        j().revalidateOnFocus && B > k && T() && (k = B + j().focusThrottleInterval, $());
      } else if (n1 == ot)
        j().revalidateOnReconnect && T() && $();
      else {
        if (n1 == it)
          return j1();
        if (n1 == Ie)
          return j1(x1);
      }
    });
    return R.current = !1, h.current = v, E.current = !0, F({
      _k: S
    }), A1 && (_(W) || _1 ? $() : Zn($)), () => {
      R.current = !0, M();
    };
  }, [
    v
  ]), V1(() => {
    let $;
    function k() {
      const M = e1(l) ? l(A().data) : l;
      M && $ !== -1 && ($ = setTimeout(Z, M));
    }
    function Z() {
      !A().error && (m || j().isVisible()) && (f || j().isOnline()) ? j1(z1).then(k) : k();
    }
    return k(), () => {
      $ && (clearTimeout($), $ = -1);
    };
  }, [
    l,
    m,
    f,
    v
  ]), wt(v1), r && _(W) && v) {
    if (!Ve && _1)
      throw new Error("Fallback data is required when using Suspense in SSR.");
    w.current = t, I.current = s, R.current = !1;
    const $ = y[v];
    if (!_($)) {
      const k = he($);
      B1(k);
    }
    if (_(g1)) {
      const k = j1(z1);
      _(v1) || (k.status = "fulfilled", k.value = !0), B1(k);
    } else
      throw g1;
  }
  return {
    mutate: he,
    get data() {
      return L.data = !0, v1;
    },
    get error() {
      return L.error = !0, g1;
    },
    get isValidating() {
      return L.isValidating = !0, pt;
    },
    get isLoading() {
      return L.isLoading = !0, gt;
    }
  };
}, fs = cs(ds), hs = () => {
}, Cs = (
  /*#__NOINLINE__*/
  hs()
), oe = Object, ke = (e) => e === Cs, ps = (e) => typeof e == "function", T1 = /* @__PURE__ */ new WeakMap(), Z1 = (e, t) => oe.prototype.toString.call(e) === `[object ${t}]`;
let gs = 0;
const ie = (e) => {
  const t = typeof e, s = Z1(e, "Date"), o = Z1(e, "RegExp"), i = Z1(e, "Object");
  let r, a;
  if (oe(e) === e && !s && !o) {
    if (r = T1.get(e), r) return r;
    if (r = ++gs + "~", T1.set(e, r), Array.isArray(e)) {
      for (r = "@", a = 0; a < e.length; a++)
        r += ie(e[a]) + ",";
      T1.set(e, r);
    }
    if (i) {
      r = "#";
      const u = oe.keys(e).sort();
      for (; !ke(a = u.pop()); )
        ke(e[a]) || (r += a + ":" + ie(e[a]) + ",");
      T1.set(e, r);
    }
  } else
    r = s ? e.toJSON() : t == "symbol" ? e.toString() : t == "string" ? JSON.stringify(e) : "" + e;
  return r;
}, ms = (e) => {
  if (ps(e))
    try {
      e = e();
    } catch {
      e = "";
    }
  const t = e;
  return e = typeof e == "string" ? e : (Array.isArray(e) ? e.length : e) ? ie(e) : "", [
    e,
    t
  ];
}, xs = (e) => ms(e ? e(0, null) : null)[0], Y1 = Promise.resolve(), ws = (e) => (t, s, o) => {
  const i = Y(!1), { cache: r, initialSize: a = 1, revalidateAll: u = !1, persistSize: d = !1, revalidateFirstPage: l = !0, revalidateOnMount: m = !1, parallel: f = !1 } = o, [, , , C] = r1.get(de);
  let c;
  try {
    c = xs(t), c && (c = lt + c);
  } catch {
  }
  const [x, p, y] = u1(r, c), v = J(() => _(x()._l) ? a : x()._l, [
    r,
    c,
    a
  ]);
  st.useSyncExternalStore(J(
    (j) => c ? y(c, () => {
      j();
    }) : () => {
    },
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [
      r,
      c
    ]
  ), v, v);
  const S = J(() => {
    const j = x()._l;
    return _(j) ? a : j;
  }, [
    c,
    a
  ]), E = Y(S());
  V1(() => {
    if (!i.current) {
      i.current = !0;
      return;
    }
    c && p({
      _l: d ? E.current : S()
    });
  }, [
    c,
    r
  ]);
  const R = m && !i.current, h = e(c, async (j) => {
    const T = x()._i, A = x()._r;
    p({
      _r: U
    });
    const F = [], z = S(), [D] = u1(r, j), L = D().data, P = [];
    let q = null;
    for (let O = 0; O < z; ++O) {
      const [b, t1] = R1(t(O, f ? null : q));
      if (!b)
        break;
      const [k1, l1] = u1(r, b);
      let W = k1().data;
      const g1 = u || T || _(W) || l && !O && !_(L) || R || L && !_(L[O]) && !o.compare(L[O], W);
      if (s && (typeof A == "function" ? A(W, t1) : g1)) {
        const m1 = async () => {
          if (!(b in C))
            W = await s(t1);
          else {
            const A1 = C[b];
            delete C[b], W = await A1;
          }
          l1({
            data: W,
            _k: t1
          }), F[O] = W;
        };
        f ? P.push(m1) : await m1();
      } else
        F[O] = W;
      f || (q = W);
    }
    return f && await Promise.all(P.map((O) => O())), p({
      _i: U
    }), F;
  }, o), w = J(
    // eslint-disable-next-line func-names
    function(j, T) {
      const A = typeof T == "boolean" ? {
        revalidate: T
      } : T || {}, F = A.revalidate !== !1;
      return c ? (F && (_(j) ? p({
        _i: !0,
        _r: A.revalidate
      }) : p({
        _i: !1,
        _r: A.revalidate
      })), arguments.length ? h.mutate(j, {
        ...A,
        revalidate: F
      }) : h.mutate()) : Y1;
    },
    // swr.mutate is always the same reference
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [
      c,
      r
    ]
  ), I = J(
    (j) => {
      if (!c) return Y1;
      const [, T] = u1(r, c);
      let A;
      if (e1(j) ? A = j(S()) : typeof j == "number" && (A = j), typeof A != "number") return Y1;
      T({
        _l: A
      }), E.current = A;
      const F = [], [z] = u1(r, c);
      let D = null;
      for (let L = 0; L < A; ++L) {
        const [P] = R1(t(L, D)), [q] = u1(r, P), O = P ? q().data : U;
        if (_(O))
          return w(z().data);
        F.push(O), D = O;
      }
      return w(F);
    },
    // exclude getKey from the dependencies, which isn't allowed to change during the lifecycle
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [
      c,
      r,
      w,
      S
    ]
  );
  return {
    size: S(),
    setSize: I,
    mutate: w,
    get data() {
      return h.data;
    },
    get error() {
      return h.error;
    },
    get isValidating() {
      return h.isValidating;
    },
    get isLoading() {
      return h.isLoading;
    }
  };
}, vs = us(fs, ws), js = /* @__PURE__ */ g("div")({
  classes: ["sfmkqcr"]
}), ys = /* @__PURE__ */ g("div")({
  classes: ["hrldtio"]
}), Es = He(({
  casts: e,
  cursor: t
}) => /* @__PURE__ */ n.jsx(js, {
  children: e.map((s, o) => /* @__PURE__ */ n.jsxs(V.Fragment, {
    children: [/* @__PURE__ */ n.jsx($1, {
      isEmbed: !1,
      ...s
    }), o < e.length - 1 && /* @__PURE__ */ n.jsx(ys, {})]
  }, s.hash))
}));
function _s(e) {
  return e ? e.map((t) => {
    var o, i, r, a;
    const s = t != null && t.replies && typeof t.replies == "object" ? Number(t.replies.count) || 0 : Number(t == null ? void 0 : t.replies) || 0;
    return {
      username: ((o = t == null ? void 0 : t.author) == null ? void 0 : o.username) ?? "",
      displayName: ((i = t == null ? void 0 : t.author) == null ? void 0 : i.display_name) ?? "",
      avatarImgUrl: ((r = t == null ? void 0 : t.author) == null ? void 0 : r.pfp_url) ?? "",
      text: (t == null ? void 0 : t.text) ?? "",
      hash: (t == null ? void 0 : t.hash) ?? "",
      reactions: (t == null ? void 0 : t.reactions) ?? [],
      replies: s,
      embeds: (t == null ? void 0 : t.embeds) ?? [],
      frames: (t == null ? void 0 : t.frames) ?? [],
      renderEmbeds: (t == null ? void 0 : t.renderEmbeds) ?? !0,
      channel: (t == null ? void 0 : t.channel) ?? "",
      viewerFid: 2,
      hasPowerBadge: ((a = t == null ? void 0 : t.author) == null ? void 0 : a.power_badge) ?? !1,
      isOwnProfile: !1,
      allowReactions: !0,
      renderFrames: !1
    };
  }) : [];
}
const Rs = (e) => p1(e).then((t) => t.json()), Ss = (e, t, s, o, i) => {
  if (t && (!t.casts || !t.casts.length)) return null;
  let r = `${C1}/v2/farcaster/feed${s ? "" : "?"}`;
  const a = new URLSearchParams();
  s && (r += `${s}?`), i && (r += `client_id=${i}`), t && (r += `&cursor=${t.next.cursor}`);
  const u = (d, l) => {
    l != null && !r.includes(`${d}=`) && a.append(d, String(l));
  };
  if (o) {
    Object.entries(o).forEach(([l, m]) => {
      const f = l.replace(/([a-z])([A-Z])/g, "$1_$2").toLowerCase();
      u(f, m);
    });
    const d = a.toString();
    d && (r += r.includes("?") ? `&${d}` : `?${d}`);
  }
  if (Object.keys(o).length > 1)
    return r;
}, mr = ({
  path: e,
  ...t
}) => {
  const {
    client_id: s
  } = o1(), o = Y(null), [i, r] = N(!1), a = Y(null), {
    data: u,
    error: d,
    size: l,
    setSize: m,
    isValidating: f
  } = vs((c, x) => Ss(c, x, e, t, s), Rs), C = f1(() => {
    if (!u) return [];
    const c = u.flatMap((p) => (p == null ? void 0 : p.casts) ?? []), x = Array.from(new Set(c.map((p) => p.hash))).map((p) => c.find((y) => y.hash === p));
    return _s(x);
  }, [u]);
  return H(() => (a.current && a.current.disconnect(), a.current = new IntersectionObserver((c) => {
    var p, y;
    const [x] = c;
    x.isIntersecting && !i && !f && u && ((y = (p = u[u.length - 1]) == null ? void 0 : p.next) != null && y.cursor) && (r(!0), m(l + 1).then(() => r(!1)));
  }, {
    rootMargin: "100px"
  }), o.current && a.current.observe(o.current), () => {
    a.current && a.current.disconnect();
  }), [o.current, i, f, u, l]), !u && !d ? /* @__PURE__ */ n.jsx("div", {
    children: "Loading..."
  }) : d ? /* @__PURE__ */ n.jsx("div", {
    children: "Error fetching feed data"
  }) : /* @__PURE__ */ n.jsxs("div", {
    children: [/* @__PURE__ */ n.jsx(Es, {
      casts: C,
      cursor: ""
    }), f && /* @__PURE__ */ n.jsxs("div", {
      style: {
        display: "flex",
        justifyContent: "center",
        padding: "10px"
      },
      children: [/* @__PURE__ */ n.jsxs("span", {
        style: {
          animation: "blink 1.5s infinite"
        },
        children: ["Loading", /* @__PURE__ */ n.jsx("span", {
          className: "dots",
          children: "..."
        })]
      }), /* @__PURE__ */ n.jsx("style", {
        children: `
                        @keyframes blink {
                            0% { opacity: 1; }
                            50% { opacity: 0; }
                            100% { opacity: 1; }
                        }
                    `
      })]
    }), /* @__PURE__ */ n.jsx("div", {
      ref: o
    })]
  });
}, As = /* @__PURE__ */ g("button")({
  classes: ["b1bqug8r"]
}), bs = /* @__PURE__ */ g("button")({
  classes: ["b1mra45w"]
}), De = "https://warpcast.com", ft = /(^|\s)\/\w+/g, ht = /@\w+/g, Ct = /((https?:\/\/)?([a-zA-Z0-9.-]+\.[a-zA-Z]{2,})(\/[^\s]*)?)/g, Fe = new RegExp(`(${ft.source})|(${ht.source})|(${Ct.source})`, "g"), Ls = (e) => ft.test(e) ? `${De}/~/channel${e.trim()}` : ht.test(e) ? `${De}/${e.substring(1)}` : Ct.test(e) ? e.startsWith("http") ? e : `http://${e}` : "", Is = /* @__PURE__ */ g("a")({
  classes: ["sby9wa2"]
}), Ts = (e) => {
  if (!e) return [];
  const t = [];
  let s = 0, o;
  for (; (o = Fe.exec(e)) !== null; ) {
    const i = o.index;
    s < i && t.push(e.slice(s, i));
    const r = Ls(o[0]), a = o[0].trim().startsWith("/");
    t.push(/* @__PURE__ */ n.jsx(Is, {
      href: r,
      target: "_blank",
      style: a ? {
        marginLeft: 3.5
      } : {},
      children: o[0].trim()
    }, i)), s = Fe.lastIndex;
  }
  return s < e.length && t.push(e.slice(s)), t;
};
function Me(e) {
  const t = Math.min(e, 1e9);
  return t >= 1e9 ? Math.floor(t / 1e8) / 10 + "B" : t >= 1e6 ? Math.floor(t / 1e5) / 10 + "M" : t >= 1e3 ? Math.floor(t / 100) / 10 + "K" : t.toString();
}
const Vs = /* @__PURE__ */ g("div")({
  classes: ["sdahm58"]
}), Ns = /* @__PURE__ */ g("div")({
  classes: ["mziqyxp"]
}), $s = /* @__PURE__ */ g("div")({
  classes: ["uz7tpmf"]
}), ks = /* @__PURE__ */ g("div")({
  classes: ["u1rxlwac"]
}), Oe = /* @__PURE__ */ g("div")({
  classes: ["pgc6yqp"]
}), Ds = /* @__PURE__ */ g("div")({
  classes: ["torjf9a"]
}), Fs = He(({
  fid: e,
  username: t,
  displayName: s,
  avatarImgUrl: o,
  bio: i,
  followers: r,
  following: a,
  hasPowerBadge: u,
  isFollowing: d,
  isOwnProfile: l,
  onCast: m,
  containerStyles: f
}) => {
  const C = Ts(i), c = f1(() => Me(a), [a]), x = f1(() => Me(r), [r]), p = () => {
    window.open("https://warpcast.com/~/settings", "_blank");
  }, y = {
    color: f == null ? void 0 : f.color
  };
  return /* @__PURE__ */ n.jsxs(Vs, {
    style: f,
    children: [l && m && /* @__PURE__ */ n.jsxs(X, {
      alignItems: "center",
      justifyContent: "space-between",
      spacingBottom: "20px",
      children: [/* @__PURE__ */ n.jsxs(ks, {
        style: f,
        children: ["@", t]
      }), /* @__PURE__ */ n.jsx(As, {
        onClick: m,
        children: "Cast"
      })]
    }), /* @__PURE__ */ n.jsxs(X, {
      children: [/* @__PURE__ */ n.jsx(K, {
        spacingRight: "10px",
        children: /* @__PURE__ */ n.jsx(N1, {
          src: o ?? E1,
          loading: "lazy",
          alt: `${s ?? "Skeleton"} Avatar`
        })
      }), /* @__PURE__ */ n.jsxs(Ns, {
        children: [/* @__PURE__ */ n.jsxs(X, {
          justifyContent: "space-between",
          flexGrow: 1,
          children: [/* @__PURE__ */ n.jsxs(Je, {
            children: [/* @__PURE__ */ n.jsxs(X, {
              children: [/* @__PURE__ */ n.jsx("strong", {
                children: s || `!${e}`
              }), u && /* @__PURE__ */ n.jsx(K, {
                spacingLeft: "5px",
                children: /* @__PURE__ */ n.jsx(Ke, {})
              })]
            }), /* @__PURE__ */ n.jsxs(X, {
              alignItems: "center",
              children: [/* @__PURE__ */ n.jsxs($s, {
                style: f,
                children: ["@", t]
              }), d && /* @__PURE__ */ n.jsx(Ds, {
                style: f,
                children: "Follows you"
              })]
            })]
          }), /* @__PURE__ */ n.jsx(X, {
            children: l && /* @__PURE__ */ n.jsx(bs, {
              style: f,
              onClick: p,
              children: "Edit Profile"
            })
          })]
        }), /* @__PURE__ */ n.jsx(K, {
          spacingVertical: "15px",
          children: /* @__PURE__ */ n.jsx("div", {
            style: f,
            children: C
          })
        }), /* @__PURE__ */ n.jsxs(X, {
          children: [/* @__PURE__ */ n.jsxs(Oe, {
            children: [/* @__PURE__ */ n.jsx("strong", {
              style: y,
              children: c
            }), " Following"]
          }), /* @__PURE__ */ n.jsxs(Oe, {
            children: [/* @__PURE__ */ n.jsx("strong", {
              style: y,
              children: x
            }), " Followers"]
          })]
        })]
      })]
    })]
  });
});
async function Ms({
  fid: e,
  viewerFid: t,
  clientId: s
}) {
  var o;
  try {
    let i = `${C1}/v2/farcaster/user/bulk?client_id=${s}&fids=${e}`;
    t && (i += `&viewer_fid=${t}`);
    const a = await (await p1(i)).json();
    return ((o = a == null ? void 0 : a.users) == null ? void 0 : o[0]) ?? null;
  } catch (i) {
    return console.error("Error fetching user by fid", i), null;
  }
}
const xr = ({
  fid: e,
  viewerFid: t,
  containerStyles: s
}) => {
  var C;
  const {
    client_id: o
  } = o1(), [i, r] = N(null), [a, u] = N(!0), [d, l] = N(null), m = (i == null ? void 0 : i.fid) === t;
  H(() => {
    e && (u(!0), l(null), Ms({
      fid: e,
      viewerFid: t,
      clientId: o
    }).then((c) => {
      r(c);
    }).catch((c) => {
      l(c);
    }).finally(() => {
      u(!1);
    }));
  }, [e, t]);
  const f = J(() => {
  }, []);
  return a ? /* @__PURE__ */ n.jsx("div", {
    children: "Loading..."
  }) : !i || d ? /* @__PURE__ */ n.jsx("div", {
    children: "Error fetching user data"
  }) : /* @__PURE__ */ n.jsx(Fs, {
    fid: e,
    username: i.username,
    displayName: i.display_name,
    avatarImgUrl: i.pfp_url,
    bio: i.profile.bio.text,
    followers: i.follower_count,
    following: i.following_count,
    hasPowerBadge: i.power_badge,
    isOwnProfile: m,
    isFollowing: (C = i.viewer_context) == null ? void 0 : C.followed_by,
    onCast: f,
    containerStyles: s
  });
}, Os = /* @__PURE__ */ g("ul")({
  classes: ["d12rx4om"]
}), Hs = /* @__PURE__ */ g("li")({
  classes: ["lrpm6lp"]
}), Us = /* @__PURE__ */ g("img")({
  classes: ["a1e9hg3b"]
}), Ps = /* @__PURE__ */ g("div")({
  classes: ["u1aqay1i"]
}), qs = /* @__PURE__ */ g("div")({
  classes: ["d1m0njq"]
}), Ws = /* @__PURE__ */ g("div")({
  classes: ["u1lvlbuo"]
}), Bs = ({
  users: e,
  onSelect: t,
  customStyles: s = {}
}) => /* @__PURE__ */ n.jsx(Os, {
  style: {
    ...s == null ? void 0 : s.dropdown
  },
  children: e.map((o) => /* @__PURE__ */ n.jsxs(Hs, {
    style: {
      ...s == null ? void 0 : s.listItem
    },
    onClick: () => t(o),
    children: [/* @__PURE__ */ n.jsx(Us, {
      style: {
        ...s == null ? void 0 : s.avatar
      },
      src: o.pfp_url,
      alt: o.username
    }), /* @__PURE__ */ n.jsxs(Ps, {
      style: {
        ...s.userInfo
      },
      children: [/* @__PURE__ */ n.jsx(qs, {
        children: o.display_name
      }), /* @__PURE__ */ n.jsxs(Ws, {
        children: ["@", o.username]
      })]
    })]
  }, o.fid))
}), zs = /* @__PURE__ */ g("div")({
  classes: ["c9hm20t"]
}), Zs = /* @__PURE__ */ g("input")({
  classes: ["i80sd8h"]
});
async function Ys({
  q: e,
  viewerFid: t,
  client_id: s
}) {
  var o;
  try {
    let i = `${C1}/v2/farcaster/user/search?q=${e}&limit=5${t ? `&viewer_fid=${t}` : ""}&client_id=${s}`;
    const a = await (await p1(i)).json();
    return ((o = a == null ? void 0 : a.result) == null ? void 0 : o.users) || [];
  } catch (i) {
    return console.log("Error fetching users by query", i), null;
  }
}
const wr = ({
  value: e,
  onChange: t,
  style: s = {},
  placeholder: o = "Enter username",
  disabled: i = !1,
  viewerFid: r,
  customStyles: a = {},
  limit: u = null
}) => {
  const {
    client_id: d
  } = o1(), [l, m] = N(""), [f, C] = N(!1), [c, x] = N([]), p = Y(null);
  H(() => {
    const h = (e == null ? void 0 : e.split(",")) || [];
    if (!h[h.length - 1]) {
      m("");
      return;
    }
    m(h[h.length - 1].trim());
  }, [e]), H(() => {
    l !== "" && !/^\d+$/.test(l) ? y(l) : C(!1);
  }, [l]);
  const y = async (h) => {
    const w = await Ys({
      q: h,
      viewerFid: r,
      client_id: d
    });
    w && (x(w), C(!0));
  }, v = (h) => {
    const w = h.target.value;
    t(w);
  }, S = (h) => {
    let w = e.split(",");
    u !== null && w.length >= u ? w[w.length - 1] = h.fid.toString() : w.push(h.fid.toString());
    const I = w.join(",");
    t(I), m(""), C(!1);
  }, E = () => {
    l && isNaN(Number(l)) && C(!0);
  }, R = () => {
    setTimeout(() => C(!1), 200);
  };
  return /* @__PURE__ */ n.jsxs(zs, {
    style: s,
    children: [/* @__PURE__ */ n.jsx(Zs, {
      ref: p,
      type: "text",
      value: e,
      onChange: v,
      onFocus: E,
      onBlur: R,
      placeholder: o,
      disabled: i
    }), f && /* @__PURE__ */ n.jsx(Bs, {
      users: c,
      onSelect: S,
      customStyles: a
    })]
  });
}, Gs = ({
  color: e = "#855DCD"
}) => /* @__PURE__ */ n.jsxs("svg", {
  xmlns: "http://www.w3.org/2000/svg",
  width: "24",
  height: "24",
  viewBox: "0 0 24 24",
  fill: "none",
  stroke: e,
  "stroke-width": "2",
  "stroke-linecap": "round",
  "stroke-linejoin": "round",
  className: "lucide lucide-search",
  children: [/* @__PURE__ */ n.jsx("circle", {
    cx: "11",
    cy: "11",
    r: "8"
  }), /* @__PURE__ */ n.jsx("path", {
    d: "m21 21-4.3-4.3"
  })]
}), Js = ({
  color: e = "#855DCD"
}) => /* @__PURE__ */ n.jsx("svg", {
  xmlns: "http://www.w3.org/2000/svg",
  width: "24",
  height: "24",
  viewBox: "0 0 24 24",
  fill: "none",
  stroke: e,
  "stroke-width": "2",
  "stroke-linecap": "round",
  "stroke-linejoin": "round",
  className: "lucide lucide-loader-circle",
  children: /* @__PURE__ */ n.jsx("path", {
    d: "M21 12a9 9 0 1 1-6.219-8.56"
  })
}), Ks = /* @__PURE__ */ g("div")({
  classes: ["ig7kcuc"]
}), Qs = /* @__PURE__ */ g("input")({
  classes: ["sh7z7k6"]
}), Xs = /* @__PURE__ */ g("div")({
  classes: ["s14ai203"]
}), er = /* @__PURE__ */ g("div")({
  classes: ["if18nd6"]
}), tr = /* @__PURE__ */ g("div")({
  classes: ["sd833lb"]
}), nr = /* @__PURE__ */ g("div")({
  classes: ["ugo586m"]
}), sr = /* @__PURE__ */ g("div")({
  classes: ["ukwty1d"]
}), rr = /* @__PURE__ */ g("div")({
  classes: ["u77olmy"]
}), or = /* @__PURE__ */ g("div")({
  classes: ["s1xvkxww"]
}), ir = /* @__PURE__ */ g("div")({
  classes: ["s1f6b1ja"]
}), ar = /* @__PURE__ */ g("div")({
  classes: ["udiufd8"]
}), cr = /* @__PURE__ */ g("div")({
  classes: ["f1b8a2h0"]
}), lr = /* @__PURE__ */ g("div")({
  classes: ["l1yq26j0"]
}), ur = /* @__PURE__ */ g("li")({
  classes: ["s18mk47k"]
}), vr = ({
  width: e,
  height: t,
  selectedUsers: s,
  onSelectUsers: o,
  ...i
}) => {
  const {
    client_id: r
  } = o1(), {
    user: a,
    isAuthenticated: u
  } = We(), [d, l] = N(""), [m, f] = N([]), [C, c] = N(!1), x = Y(null);
  H(() => {
    function h(w) {
      x.current && !x.current.contains(w.target) && (f([]), l(""));
    }
    return document.addEventListener("mousedown", h), () => {
      document.removeEventListener("mousedown", h);
    };
  }, []);
  const v = J(((h, w) => {
    let I;
    return (...j) => {
      clearTimeout(I), I = setTimeout(() => h(...j), w);
    };
  })(async (h, w) => {
    var I;
    if (!h) {
      f([]);
      return;
    }
    c(!0);
    try {
      const T = await (await p1(w)).json();
      f(((I = T == null ? void 0 : T.result) == null ? void 0 : I.users) || []);
    } catch (j) {
      console.error("Error fetching data:", j);
    } finally {
      c(!1);
    }
  }, 500), []), S = (h) => {
    const w = `${C1}/v2/farcaster/user/search?q=${h.target.value}&client_id=${r}${u ? `&viewer_fid=${a == null ? void 0 : a.fid}` : ""})`;
    l(h.target.value), v(h.target.value, w);
  }, E = (h) => {
    s.some((w) => w.username === h.username) || (o([...s, h]), l(""), f([]));
  }, R = (h) => {
    o(s.filter((w) => w.username !== h));
  };
  return /* @__PURE__ */ n.jsxs(Ks, {
    ref: x,
    style: {
      width: e
    },
    children: [/* @__PURE__ */ n.jsxs(er, {
      children: [s.map((h) => /* @__PURE__ */ n.jsxs(tr, {
        children: [/* @__PURE__ */ n.jsx(N1, {
          src: h.pfp_url ?? E1,
          alt: `${h.username} Avatar`,
          style: {
            width: "24px",
            height: "24px",
            borderRadius: "50%"
          }
        }), /* @__PURE__ */ n.jsx("span", {
          style: {
            marginLeft: "8px"
          },
          children: h.display_name
        }), /* @__PURE__ */ n.jsx(et, {
          size: 20,
          onClick: () => R(h.username)
        })]
      }, h.username)), /* @__PURE__ */ n.jsxs(or, {
        children: [/* @__PURE__ */ n.jsx(ir, {
          children: /* @__PURE__ */ n.jsx(Gs, {
            color: "#855DCD"
          })
        }), /* @__PURE__ */ n.jsx(Qs, {
          ...i,
          value: d,
          onChange: S,
          placeholder: "Type to search users",
          style: {
            width: e || "100%",
            height: t || "40px"
          }
        })]
      })]
    }), C ? /* @__PURE__ */ n.jsx(lr, {
      children: /* @__PURE__ */ n.jsx(Js, {
        color: "#855DCD"
      })
    }) : m.length !== 0 && /* @__PURE__ */ n.jsx(Xs, {
      children: m.map((h) => /* @__PURE__ */ n.jsxs(ur, {
        onClick: () => E(h),
        children: [/* @__PURE__ */ n.jsx(N1, {
          src: h.pfp_url ?? E1,
          alt: `${h.username} Avatar`,
          style: {
            width: "40px",
            height: "40px",
            borderRadius: "50%"
          }
        }), /* @__PURE__ */ n.jsxs(nr, {
          children: [/* @__PURE__ */ n.jsx(sr, {
            children: h.display_name
          }), /* @__PURE__ */ n.jsxs(ar, {
            children: [/* @__PURE__ */ n.jsxs(rr, {
              children: ["@", h.username]
            }), h.viewer_context && (h.viewer_context.following || h.viewer_context.followed_by) && /* @__PURE__ */ n.jsx(cr, {
              children: h.viewer_context.following && h.viewer_context.followed_by ? "Following each other" : h.viewer_context.following ? "Following" : h.viewer_context.followed_by ? "Follows you" : null
            })]
          })]
        })]
      }, h.username))
    })]
  });
};
export {
  $1 as CastCard,
  Vt as NeynarAuthButton,
  vr as NeynarAutoCompleteUser,
  qt as NeynarCastCard,
  pr as NeynarContextProvider,
  gr as NeynarConversationList,
  mr as NeynarFeedList,
  Rn as NeynarFrameCard,
  xr as NeynarProfileCard,
  wr as NeynarUserDropdown,
  c1 as SIWN_variant,
  Pe as Theme,
  le as useLocalStorage,
  o1 as useNeynarContext
};
